---
title: "Phytoplankton paper, 16S"


output: github_document
---
  

```{r global_options, echo=FALSE, results='hide'}
library(knitr)
knitr::opts_chunk$set(fig.width = 12, fig.height = 8, fig.align = 'center', results = 'hide', echo = FALSE, warning = FALSE, message = FALSE, cache = FALSE)
```


```{r load data}

library(phyloseq)
library(microViz)
library(dplyr)
library(stringr)

# import data
Sven_phyto <- readRDS("")

# remove unwanted samples
Sven_phyto = ps_filter(Sven_phyto, Culture_type != "DELETE")

# import revised metadata
new_sample_data <- read.csv(file = "", header = TRUE, fileEncoding = "UTF-8-BOM", check.names = FALSE, row.names = 1)

new_sample_data = new_sample_data %>% 
  mutate(BacterialStatus = if_else(str_detect(Culture_type, "Bacteria"), "Present", "Absent"))

new_sample_data$ExperimentalPhase = factor(new_sample_data$Cultivation_Stage,
                                       levels = c("Lag",
                                                  "Exponential",
                                                  "Stationary",
                                                  "Death"))

# update metadata
sample_data(Sven_phyto) <- new_sample_data
Sven_phyto_metadata <- data.frame(sample_data(Sven_phyto))
```

Rarefaction curves to  determine subsampling level. 30000 reads chosen. All samples used for analysis. 

```{r rarefaction}

# remove any samples with less than 100 sequences

pruned <- prune_samples(sample_sums(Sven_phyto) >= 100, Sven_phyto)
pruned

library(ggplot2)
library(ranacapa)

# calculate rarefaction curves, then plot and color by subsite
p <- ggrare(pruned, step = 1000, color = "Culture_type", se =
              FALSE,parallel = TRUE)+
  scale_y_log10() +
  scale_x_continuous(limits = c(1, 170000))
p


p + facet_wrap(~Culture_type)

# rarefy at 30k

Sven_phyto_rare <- rarefy_even_depth(Sven_phyto, 30000, rngseed=TRUE)
```

```{r theme}

My_Theme = theme_bw() +
  theme(axis.title.x = element_text(size=18),
        axis.text.x = element_text(angle=0, colour = "black", vjust=1, hjust = 0.5, size=18), 
        axis.text.y = element_text(colour = "black", size=18),
        axis.title.y = element_text(size=18),
        plot.title = element_text(size = 18),
        legend.title =element_text(size = 18),
        legend.text = element_text(size = 18),
        legend.position="right",
        legend.key.size = unit(1, "cm"),
        strip.text.x = element_text(size=18),
        strip.text.y = element_text(size=18),
        #panel.background = element_blank(),
        panel.border = element_rect(fill = NA, colour = "black"),
        strip.background = element_rect(colour="black"))


# culture_type colours
Cond.col <- c("None" = "black",
              "Coccolithophore" = "#D83706FF",
              "Diatom" = "#117733")

Cond.sha = c("BacterialControl" = 1, #Hollow circle
             "CoccolithophoreBacteria" = 1, #Hollow circle
             "CoccolithophoreOnly" = 16, #Filled circle
             "DiatomBacteria" = 1, #Hollow circle
             "DiatomOnly" = 16) #Filled circle

BactStat.sha = c("Present" = 16, #Filled circle 
            "Absent" = 1) #Hollow circle

```

```{r Calculate alpha diversity [Richness, Shannon, Evenness]}

alpha_summary_SPR<- estimate_richness(Sven_phyto_rare, measures = c("Observed", "Shannon"))

# this calculates eveness and adds it to the prior file

Evenness <- microbiome::evenness(Sven_phyto_rare, 'pielou')
alpha_summary_SPR$Pielou <- Evenness$pielou

# combine results with sample_data. This is useful if we wish to plot later against other variables or information.
alpha_meta_SPR <- data.frame(alpha_summary_SPR, sample_data(Sven_phyto_rare))




library(tidyr)
library(Rmisc)

# convert data from wide to long for easier plotting
alpha_meta_SPR_long = gather(alpha_meta_SPR, Alpha, alpha_value, c("Observed", "Shannon", "Pielou"), factor_key = FALSE)

# summarise data
Observed_sum.df = summarySE(alpha_meta_SPR_long, measurevar = "alpha_value", groupvars = c("Alpha","Culture_type", "Phyto", "HoursFromStart", "Cultivation_Stage", "BacterialStatus"))

Observed_sum.df$ExperimentalPhase = factor(Observed_sum.df$Cultivation_Stage,
                                       levels = c("Lag",
                                                  "Exponential",
                                                  "Stationary",
                                                  "Death"))

Observed_sum.df$Phyto<-as.factor(Observed_sum.df$Phyto)
  
  
alpha_plot <- ggplot(Observed_sum.df, aes(x = HoursFromStart, 
                                     y = alpha_value, 
                                     color = Phyto, 
                                     shape = BacterialStatus,
                                     group = Culture_type))+
  geom_point(size = 3)+
  geom_line() +
  geom_errorbar(aes(ymin=alpha_value-sd, 
                    ymax=alpha_value+sd), 
                width = 0.2) +
  theme_bw() +
  scale_shape_manual("Microbiome", values = BactStat.sha)+
  scale_color_manual("Phytoplankton",values = Cond.col) +
  ylab("Alpha diversity") +
  xlab("Hours") +
  My_Theme +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  facet_grid(Alpha~., scales = "free_y")
alpha_plot 

ggsave("",width = 6,height = 10,units ="in", device="pdf")

alpha_plot_stage <- ggplot(Observed_sum.df, aes(x = Cultivation_Stage, 
                                     y = alpha_value, 
                                     color = Phyto, 
                                     shape = BacterialStatus,
                                     group = Culture_type))+
  geom_point(size = 3)+
  geom_line() +
  geom_errorbar(aes(ymin=alpha_value-sd, 
                    ymax=alpha_value+sd), 
                width = 0.2) +
  theme_bw() +
  scale_shape_manual("Microbiome", values = BactStat.sha)+
  scale_color_manual("Phytoplankton",values = Cond.col) +
  ylab("Alpha diversity") +
  xlab("Growth phase") +
  My_Theme+
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  facet_grid(Alpha~., scales = "free_y") +
  scale_x_discrete(limits = c("Lag","Exponential","Stationary","Death"))
alpha_plot_stage 

ggsave("", width = 6, height = 10,units ="in", device="pdf")

```



NMDS ordinations based on 16s data
ANOSIM and ADONIS results shown below done by Culture_type

```{r Compare beta diversity, results='markup'}

# ordinate data using Non-metric multidimensional scaling (NMDS) on Bray–Curtis dissimilarity (distances)
set.seed(1)
NMDS_16S.ord <- ordinate(Sven_phyto_rare, "NMDS", "bray")

# by calling the newly created file we can get the stress value of the plot
NMDS_16S.ord

library(vegan)
library(grid)
library(ggforce)

stressplot(NMDS_16S.ord)

# create label for stress value
Stress_val = grobTree(textGrob("Stress = 0.18", x = 0.05,  y = 0.05, hjust = 0, gp = gpar(col = "Black", fontsize = 14, fontface = "italic")))


Sven_phyto_16S_NMDS <- plot_ordination(Sven_phyto_rare, NMDS_16S.ord, type = "samples",color = "Phyto", shape = "BacterialStatus") +
  geom_point(size = 3) +
  scale_shape_manual("Microbiome", values = BactStat.sha)+
  scale_color_manual("Phytoplankton",values = Cond.col) +
  My_Theme +
  annotation_custom(Stress_val) +
  theme_bw(base_size = 12) +
  geom_mark_ellipse(aes(group = Cultivation_Stage, label = Cultivation_Stage,linetype = Cultivation_Stage),alpha = 0.2, con.cap = 0) +
xlim(-0.7, 0.7) +
ylim(-0.6, 0.6) +
scale_linetype_manual(breaks=c("Lag", "Exponential", "Stationary", "Death"),values=c("solid", "longdash", "dotdash", "dotted")) +
    scale_linetype_discrete(name = "Growth Stage")+
  ggtitle("All") +
  theme(
    plot.title=element_text(face = "bold")
  )

Sven_phyto_16S_NMDS$layers <- Sven_phyto_16S_NMDS$layers[-1]

Sven_phyto_16S_NMDS

ggsave("",width = 7,height = 5, units ="in", device = "pdf")


# stats by Culture_type
# ANOSIM
Culture_type_group = get_variable(Sven_phyto_rare, "Culture_type")
Culture_type_ano = anosim(phyloseq::distance(Sven_phyto_rare, "bray"), Culture_type_group)
Culture_type_ano$signif
Culture_type_ano$statistic

# ADONIS
# create a data frame using your sample_data
df_ado = as(sample_data(Sven_phyto_rare), "data.frame")
# calculate your Bray distance matrix
Culture_type_ado = phyloseq::distance(Sven_phyto_rare, "bray")
# perform your ADONIS test
Culture_type_ado_stats <- adonis2(Culture_type_ado ~ Culture_type, df_ado)

# check results
Culture_type_ado_stats


# stats by Growth stage
# ANOSIM
Cultivation_Stage_group = get_variable(Sven_phyto_rare, "Cultivation_Stage")
Cultivation_Stage_ano = anosim(phyloseq::distance(Sven_phyto_rare, "bray"), Cultivation_Stage_group)
Cultivation_Stage_ano$signif
Cultivation_Stage_ano$statistic

# ADONIS
# create a data frame using your sample_data
df_ado = as(sample_data(Sven_phyto_rare), "data.frame")
# calculate your Bray distance matrix
Cultivation_Stage_ado = phyloseq::distance(Sven_phyto_rare, "bray")
# perform your ADONIS test
Cultivation_Stage_ado_stats <- adonis2(Cultivation_Stage_ado ~ Culture_type, df_ado)

# check results
Cultivation_Stage_ado_stats


```


```{r Compare beta diversity Lag, results='markup'}

Sven_phyto_rare_Lag <- subset_samples(Sven_phyto_rare, Cultivation_Stage == "Lag")
Sven_phyto_rare_Lag

# remove outlier point

Sven_phyto_rare_Lag <- subset_samples(Sven_phyto_rare_Lag, sample_names(Sven_phyto_rare_Lag)!="PP2B_1_T0")

# ordinate data using Non-metric multidimensional scaling (NMDS) on Bray–Curtis dissimilarity (distances)
set.seed(1)
Lag_NMDS_16S <- ordinate(Sven_phyto_rare_Lag, "NMDS", "bray")

# by calling the newly created file we can get the stress value of the plot
Lag_NMDS_16S
nmds_coordinates <- Lag_NMDS_16S$points

stressplot(Lag_NMDS_16S)

# create label for stress value
Lag_Stress_val = grobTree(textGrob("Stress = 0.13", x = 0.05,  y = 0.05, hjust = 0, gp = gpar(col = "Black", fontsize = 14,fontface = "italic")))

Lag_Sven_phyto_16S_NMDS <- plot_ordination(Sven_phyto_rare_Lag, Lag_NMDS_16S, type = "samples",color = "Phyto", shape = "BacterialStatus") +
  geom_point(size = 3) +
  scale_shape_manual("Microbiome", values = BactStat.sha)+
  scale_color_manual("Phytoplankton",values = Cond.col) +
  My_Theme+
  annotation_custom(Lag_Stress_val) +
  theme_bw(base_size=12) +
  scale_linetype_manual(breaks=c("Lag", "Lag", "Stationary", "Death"),values=c("solid", "longdash", "dotdash", "dotted"))+
  scale_linetype_discrete(name = "Phytoplankton")+
  geom_mark_ellipse(aes(group=Phyto, label = Phyto,linetype = Phyto),alpha = 1,con.cap = 1) +
  xlim(-1, 1) +
  ylim(-1.1, 1.2) +
  ggtitle("Lag") +
  theme(
    plot.title=element_text(face='bold')
  )

Lag_Sven_phyto_16S_NMDS$layers <- Lag_Sven_phyto_16S_NMDS$layers[-1]

Lag_Sven_phyto_16S_NMDS



ggsave("", width = 7, height = 5,units = "in", device = "pdf")


# stats by Culture_type
# ANOSIM
Culture_type_group = get_variable(Sven_phyto_rare_Lag, "Culture_type")
Culture_type_ano = anosim(phyloseq::distance(Sven_phyto_rare_Lag, "bray"), Culture_type_group)
Culture_type_ano$signif
Culture_type_ano$statistic

# ADONIS
# create a data frame using your sample_data
df_ado = as(sample_data(Sven_phyto_rare_Lag), "data.frame")
# calculate your Bray distance matrix
Culture_type_ado = phyloseq::distance(Sven_phyto_rare_Lag, "bray")
# perform your ADONIS test
Culture_type_ado_stats <- adonis2(Culture_type_ado ~ Culture_type, df_ado)

# check results
Culture_type_ado_stats

```


```{r Compare beta diversity Exponential, results='markup'}

Sven_phyto_rare_Exponential <- subset_samples(Sven_phyto_rare, Cultivation_Stage == "Exponential")
Sven_phyto_rare_Exponential

# Ordinate data using Non-metric multidimensional scaling (NMDS) on Bray–Curtis dissimilarity (distances)
set.seed(1)
Exponential_NMDS_16S <- ordinate(Sven_phyto_rare_Exponential, "NMDS", "bray")

# By calling the newly created file we can get the stress value of the plot
Exponential_NMDS_16S
nmds_coordinates <- Exponential_NMDS_16S$points

stressplot(Exponential_NMDS_16S)

# Create label for stress value
Exponential_Stress_val = grobTree(textGrob("Stress = 0.12", x = 0.04,  y = 0.03, hjust = 0, gp = gpar(col = "Black", fontsize = 14, fontface = "italic")))


Exponential_Sven_phyto_16S_NMDS <- plot_ordination(Sven_phyto_rare_Exponential, Exponential_NMDS_16S, type = "samples",color = "Phyto", shape= "BacterialStatus")+
  geom_point(size = 3)+
  scale_shape_manual("Microbiome", values = BactStat.sha) +
  scale_color_manual("Phytoplankton",values = Cond.col) +
  My_Theme+
  annotation_custom(Exponential_Stress_val) +
  theme_bw(base_size = 12) +
  scale_linetype_manual(breaks=c("Exponential", "Exponential", "Stationary", "Death"),values=c("solid", "longdash", "dotdash", "dotted")) +
  scale_linetype_discrete(name = "Phytoplankton") +
  geom_mark_ellipse(aes(group=Phyto, label = Phyto,linetype = Phyto),alpha = 1,con.cap = 1) +
  xlim(-1.3, 1.2) +
  ylim(-1.5, 1.4) +
  ggtitle("Exponential") +
  theme(
    plot.title=element_text(face='bold')
  )

Exponential_Sven_phyto_16S_NMDS$layers <- Exponential_Sven_phyto_16S_NMDS$layers[-1]

Exponential_Sven_phyto_16S_NMDS



ggsave("", width = 7, height = 5,units = "in", device = "pdf")


# Stats by Culture_type
# ANOSIM
Culture_type_group = get_variable(Sven_phyto_rare_Exponential, "Culture_type")
Culture_type_ano = anosim(phyloseq::distance(Sven_phyto_rare_Exponential, "bray"), Culture_type_group)
Culture_type_ano$signif
Culture_type_ano$statistic

# ADONIS
# Create a data frame using your sample_data
df_ado = as(sample_data(Sven_phyto_rare_Exponential), "data.frame")
# Calculate your Bray distance matrix
Culture_type_ado = phyloseq::distance(Sven_phyto_rare_Exponential, "bray")
# Perform your ADONIS test
Culture_type_ado_stats <- adonis2(Culture_type_ado ~ Culture_type, df_ado)

# Check results
Culture_type_ado_stats

```


```{r Compare beta diversity Stationary, results='markup'}

Sven_phyto_rare_Stationary <- subset_samples(Sven_phyto_rare, Cultivation_Stage == "Stationary")
Sven_phyto_rare_Stationary

# Ordinate data using Non-metric multidimensional scaling (NMDS) on Bray–Curtis dissimilarity (distances)
set.seed(1)
Stationary_NMDS_16S <- ordinate(Sven_phyto_rare_Stationary, "NMDS", "bray")

# By calling the newly created file we can get the stress value of the plot
Stationary_NMDS_16S

nmds_coordinates <- Stationary_NMDS_16S$points


stressplot(Stationary_NMDS_16S)

# Create label for stress value
Stationary_Stress_val = grobTree(textGrob("Stress = 0.16", x = 0.05,  y = 0.05, hjust = 0, gp = gpar(col = "Black", fontsize = 14,fontface = "italic")))


Stationary_Sven_phyto_16S_NMDS <- plot_ordination(Sven_phyto_rare_Stationary, Stationary_NMDS_16S, type = "samples", color = "Phyto", shape = "BacterialStatus") +
  geom_point(size=3) +
  scale_shape_manual("Microbiome", values = BactStat.sha) +
  scale_color_manual("Phytoplankton",values = Cond.col) +
  My_Theme +
  annotation_custom(Stationary_Stress_val) +
  theme_bw(base_size=12) +
  scale_linetype_manual(breaks=c("Stationary", "Stationary", "Stationary", "Death"), values = c("solid", "longdash", "dotdash", "dotted")) +
  scale_linetype_discrete(name = "Phytoplankton") +
  geom_mark_ellipse(aes(group=Phyto, label = Phyto,linetype = Phyto), alpha = 1,con.cap = 1)+
  xlim(-2.1, 1.9) +
  ylim(-2.1, 1.8) +
  ggtitle("Stationary") +
  theme(
    plot.title = element_text(face = 'bold')
  )

Stationary_Sven_phyto_16S_NMDS$layers <- Stationary_Sven_phyto_16S_NMDS$layers[-1]

Stationary_Sven_phyto_16S_NMDS



ggsave("", width = 7, height = 5, units = "in", device = "pdf")


# stats by Culture_type
# ANOSIM
Culture_type_group = get_variable(Sven_phyto_rare_Stationary, "Culture_type")
Culture_type_ano = anosim(phyloseq::distance(Sven_phyto_rare_Stationary, "bray"), Culture_type_group)
Culture_type_ano$signif
Culture_type_ano$statistic

# ADONIS
# create a data frame using your sample_data
df_ado = as(sample_data(Sven_phyto_rare_Stationary), "data.frame")
# calculate your Bray distance matrix
Culture_type_ado = phyloseq::distance(Sven_phyto_rare_Stationary, "bray")
# perform your ADONIS test
Culture_type_ado_stats <- adonis2(Culture_type_ado ~ Culture_type, df_ado)

# check results
Culture_type_ado_stats

```


```{r Compare beta diversity Death, results='markup'}

Sven_phyto_rare_Death <- subset_samples(Sven_phyto_rare, Cultivation_Stage == "Death")
Sven_phyto_rare_Death

# ordinate data using Non-metric multidimensional scaling (NMDS) on Bray–Curtis dissimilarity (distances)
set.seed(1)
Death_NMDS_16S <- ordinate(Sven_phyto_rare_Death, "NMDS", "bray")

# by calling the newly created file we can get the stress value of the plot
Death_NMDS_16S

nmds_coordinates <- Death_NMDS_16S$points


stressplot(Death_NMDS_16S)

# create label for stress value
Death_Stress_val = grobTree(textGrob("Stress = 0.15", x = 0.05,  y = 0.05, hjust = 0, gp = gpar(col = "Black", fontsize = 14, fontface = "italic")))


Death_Sven_phyto_16S_NMDS <- plot_ordination(Sven_phyto_rare_Death, Death_NMDS_16S, type = "samples", color ="Phyto", shape = "BacterialStatus") +
  geom_point(size = 3) +
  scale_shape_manual("Microbiome", values = BactStat.sha)+
  scale_color_manual("Phytoplankton",values = Cond.col) +
  My_Theme+
  annotation_custom(Death_Stress_val) +
  theme_bw(base_size=12) +
  scale_linetype_manual(breaks = c("Death", "Death", "Death", "Death"),values=c("solid", "longdash", "dotdash", "dotted"))+
  scale_linetype_discrete(name = "Phytoplankton") +
  geom_mark_ellipse(aes(group = Phyto, label = Phyto,linetype = Phyto), alpha = 1, con.cap = 1)+
  xlim(-1, 1.3) +
  ylim(-1.3, 1.3) +
  ggtitle("Death") +
  theme(
    plot.title=element_text(face = 'bold')
  )

Death_Sven_phyto_16S_NMDS$layers <- Death_Sven_phyto_16S_NMDS$layers[-1]

Death_Sven_phyto_16S_NMDS



ggsave("", width = 7, height = 5, units = "in", device = "pdf")


# stats by Culture_type
# ANOSIM
Culture_type_group = get_variable(Sven_phyto_rare_Death, "Culture_type")
Culture_type_ano = anosim(phyloseq::distance(Sven_phyto_rare_Death, "bray"), Culture_type_group)
Culture_type_ano$signif
Culture_type_ano$statistic

# ADONIS
# create a data frame using your sample_data
df_ado = as(sample_data(Sven_phyto_rare_Death), "data.frame")
# calculate your Bray distance matrix
Culture_type_ado = phyloseq::distance(Sven_phyto_rare_Death, "bray")
# perform your ADONIS test
Culture_type_ado_stats <- adonis2(Culture_type_ado ~ Culture_type, df_ado)

# check results
Culture_type_ado_stats

```

```{r Combine NMDS plots}

Combined_NMDS_plot = ggpubr::ggarrange(Lag_Sven_phyto_16S_NMDS,
                                  Exponential_Sven_phyto_16S_NMDS,
                                  Stationary_Sven_phyto_16S_NMDS,
                                  Death_Sven_phyto_16S_NMDS,
                                  ncol = 2,
                                  nrow = 2,
                                  common.legend = T,
                                  legend = "right")


pdf("", width = 15, height = 15)
Combined_NMDS_plot
dev.off()



Combined_NMDS_plot_all =ggpubr::ggarrange(Sven_phyto_16S_NMDS,Combined_NMDS_plot, widths = c(2,2),heights = c(1, 2))
pdf("", width = 19, height = 10)
Combined_NMDS_plot_all
dev.off()
```


```{r distance comparisons Lag}

Totu_table = t(otu_table(Sven_phyto_rare_Lag)) # transpose otu table
otu_table(Sven_phyto_rare_Lag) = Totu_table

p = Sven_phyto_rare_Lag
m = "bray"
s = "Original.names"
d = "Culture_type"

# calculate distances
library(reshape2)
library(reshape)
library(dplyr)
wu = vegan::vegdist(t(otu_table(p)), method = "bray")
wu.m = melt(as.matrix(wu))

colnames(wu.m) <- c("Var1", "Var2", "value")


# remove self-comparisons
wu.m = wu.m %>%
  filter(as.character(Var1) != as.character(Var2)) %>%
  mutate_if(is.factor, as.character)

# get sample data (S4 error OK and expected)
sd = data.frame(sample_data(p))%>%
  select(s, d)%>%
  mutate_if(is.factor, as.character) 

library(tibble)
sd <- tibble::rownames_to_column(sd, "Sample.names")
sd = subset(sd, select = -c(Original.names) )


# combined distances with sample data
colnames(sd) = c("Var1", "Culture_type")
wu.sd = left_join(wu.m, sd, by = "Var1")

colnames(sd) = c("Var2", "Culture_type2")
wu.sd = left_join(wu.sd, sd, by = "Var2")

wu.sd.sums = wu.sd %>%
  filter(Culture_type == 'BacterialControl') %>%
  mutate_if(is.factor,as.character)

# order factors
wu.sd.sums$Culture_type2 = factor(wu.sd.sums$Culture_type2, levels = c('BacterialControl', 'CoccolithophoreOnly', 'CoccolithophoreBacteria' ,'DiatomOnly','DiatomBacteria'))

library(ggpubr)

sdpb = data.frame(sample_data(p))%>%
  select(s, d, Phyto,BacterialStatus)%>%
  mutate_if(is.factor, as.character)

sdpb <- tibble::rownames_to_column(sdpb, "Sample.names")
sdpb = subset(sdpb, select = -c(Original.names) )
colnames(sdpb) = c("Var2", "Culture_type2","Phyto", "BacterialStatus")
wu.sd.sums = left_join(wu.sd.sums, sdpb, by = "Var2")

wu.sd.sums$Phyto<-as.factor(wu.sd.sums$Phyto)

Lag_16S_dist_plot<-ggplot(wu.sd.sums, aes(x = Culture_type2.x, y = value,
                                                 fill = Phyto, 
                                                 group = Culture_type2.x)) +
  geom_boxplot() +
  geom_point() +
  ylab("Bray distance") + xlab("") +
  stat_compare_means(method = "kruskal.test", size = 3, label.x = 3, label.y = .2, aes(group = Culture_type2.x)) +
  stat_compare_means(aes(label = after_stat(p.signif)),
                     method = "t.test", ref.group = "BacterialControl") + 
  ggtitle("Lag") +
  theme_bw() +
  scale_shape_manual("Microbiome", values = BactStat.sha)+
  scale_fill_manual("Phytoplankton",values = Cond.col) +
  ylab("Bray Distance") +
  My_Theme +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  scale_y_continuous(expand = expansion(mult = c(0.1, 0.2))) +
  scale_x_discrete(breaks=c('BacterialControl', 'CoccolithophoreOnly', 'CoccolithophoreBacteria' ,'DiatomOnly','DiatomBacteria'),
                     labels=c("Control", "Coccolithophore", "Coccolithophore + microbiome", "Diatom", "Diatom + microbiome")) +
  theme(
    plot.title=element_text(face = "bold")
  )

Lag_16S_dist_plot
ggsave("", width = 4, height = 4, units = "in", device = "pdf")



```

```{r distance comparisons Exponential}

Sven_phyto_rare_Exponential <- subset_samples(Sven_phyto_rare, Cultivation_Stage == "Exponential")
Sven_phyto_rare_Exponential

Totu_table = t(otu_table(Sven_phyto_rare_Exponential)) # transpose otu table
otu_table(Sven_phyto_rare_Exponential) = Totu_table

p = Sven_phyto_rare_Exponential
m = "bray"
s = "Original.names"
d = "Culture_type"

# Calculate distances
library(reshape2)
library(reshape)
library(dplyr)
wu = vegan::vegdist(t(otu_table(p)), method = "bray")
wu.m = melt(as.matrix(wu))

colnames(wu.m) <- c("Var1", "Var2", "value")

# remove self-comparisons
wu.m = wu.m %>%
  filter(as.character(Var1) != as.character(Var2)) %>%
  mutate_if(is.factor, as.character)

# get sample data (S4 error OK and expected)
sd = data.frame(sample_data(p))%>%
  select(s, d)%>%
  mutate_if(is.factor, as.character) 

library(tibble)
sd <- tibble::rownames_to_column(sd, "Sample.names")
sd = subset(sd, select = -c(Original.names) )


# combined distances with sample data
colnames(sd) = c("Var1", "Culture_type")
wu.sd = left_join(wu.m, sd, by = "Var1")

colnames(sd) = c("Var2", "Culture_type2")
wu.sd = left_join(wu.sd, sd, by = "Var2")

wu.sd.sums = wu.sd %>%
  filter(Culture_type == 'BacterialControl') %>%
  mutate_if(is.factor,as.character)

# order factors
wu.sd.sums$Culture_type2 = factor(wu.sd.sums$Culture_type2, levels = c('BacterialControl', 'CoccolithophoreOnly', 'CoccolithophoreBacteria' ,'DiatomOnly','DiatomBacteria'))



sdpb = data.frame(sample_data(p))%>%
  select(s, d, Phyto,BacterialStatus)%>%
  mutate_if(is.factor, as.character)

sdpb <- tibble::rownames_to_column(sdpb, "Sample.names")
sdpb = subset(sdpb, select = -c(Original.names) )
colnames(sdpb) = c("Var2", "Culture_type2","Phyto", "BacterialStatus")
wu.sd.sums = left_join(wu.sd.sums, sdpb, by = "Var2")


wu.sd.sums$BacterialStatus = factor(wu.sd.sums$BacterialStatus,
                                           levels = c("None",
                                                      "Coccolithophore",
                                                      "Diatom"))

wu.sd.sums$Phyto<-as.factor(wu.sd.sums$Phyto)



Exponential_16S_dist_plot<-ggplot(wu.sd.sums, aes(x = Culture_type2.x, y = value,
                                                 fill = Phyto, 
                                                 group = Culture_type2.x))+
  geom_boxplot() +
  geom_point()+
  ylab("Bray distance") + xlab("") +
  stat_compare_means(method = "kruskal.test", size = 3, label.x = 3, label.y = .5, aes(group = Culture_type2.x))+
  stat_compare_means(aes(label = after_stat(p.signif)),
                     method = "t.test", ref.group = "BacterialControl") + 
  ggtitle("Exponential")+
  theme_bw()+
  scale_shape_manual("Microbiome", values = BactStat.sha)+
  scale_fill_manual("Phytoplankton",values = Cond.col) +
  ylab("Bray Distance")+
  My_Theme+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  scale_y_continuous(expand = expansion(mult = c(0.1, 0.2)))+
  scale_x_discrete(breaks=c('BacterialControl', 'CoccolithophoreOnly', 'CoccolithophoreBacteria' ,'DiatomOnly','DiatomBacteria'),
                     labels=c("Control","Coccolithophore","Coccolithophore + microbiome", "Diatom", "Diatom + microbiome"))+
  theme(
    plot.title=element_text(face='bold')
  )


Exponential_16S_dist_plot
ggsave("", width = 4, height = 4, units = "in", device = "pdf")



```

```{r distance comparisons Stationary}

Sven_phyto_rare_Stationary <- subset_samples(Sven_phyto_rare, Cultivation_Stage == "Stationary")
Sven_phyto_rare_Stationary

Totu_table = t(otu_table(Sven_phyto_rare_Stationary)) # transpose otu table
otu_table(Sven_phyto_rare_Stationary) = Totu_table

p = Sven_phyto_rare_Stationary
m = "bray"
s = "Original.names"
d = "Culture_type"

# calculate distances

wu = vegan::vegdist(t(otu_table(p)), method = "bray")
wu.m = melt(as.matrix(wu))

colnames(wu.m) <- c("Var1", "Var2", "value")

# remove self-comparisons
wu.m = wu.m %>%
  filter(as.character(Var1) != as.character(Var2)) %>%
  mutate_if(is.factor, as.character)

# Get sample data (S4 error OK and expected)
sd = data.frame(sample_data(p))%>%
  select(s, d)%>%
  mutate_if(is.factor, as.character) 


sd <- tibble::rownames_to_column(sd, "Sample.names")
sd = subset(sd, select = -c(Original.names) )


# combined distances with sample data
colnames(sd) = c("Var1", "Culture_type")
wu.sd = left_join(wu.m, sd, by = "Var1")

colnames(sd) = c("Var2", "Culture_type2")
wu.sd = left_join(wu.sd, sd, by = "Var2")

wu.sd.sums = wu.sd %>%
  filter(Culture_type == 'BacterialControl') %>%
  mutate_if(is.factor,as.character)

# order factors
wu.sd.sums$Culture_type2 = factor(wu.sd.sums$Culture_type2, levels = c('BacterialControl', 'CoccolithophoreOnly', 'CoccolithophoreBacteria' ,'DiatomOnly','DiatomBacteria'))


sdpb = data.frame(sample_data(p))%>%
  select(s, d, Phyto,BacterialStatus)%>%
  mutate_if(is.factor, as.character)

sdpb <- tibble::rownames_to_column(sdpb, "Sample.names")
sdpb = subset(sdpb, select = -c(Original.names) )
colnames(sdpb) = c("Var2", "Culture_type2","Phyto", "BacterialStatus")
wu.sd.sums = left_join(wu.sd.sums, sdpb, by = "Var2")


wu.sd.sums$BacterialStatus = factor(wu.sd.sums$BacterialStatus,
                                           levels = c("None",
                                                      "Coccolithophore",
                                                      "Diatom"))

wu.sd.sums$Phyto<-as.factor(wu.sd.sums$Phyto)



Stationary_16S_dist_plot<-ggplot(wu.sd.sums, aes(x = Culture_type2.x, y = value,
                                                 fill = Phyto, 
                                                 group = Culture_type2.x))+
  geom_boxplot() +
  geom_point() +
  ylab("Bray distance") + xlab("") +
  stat_compare_means(method = "kruskal.test", size = 3,label.x = 3, label.y = .6, aes(group = Culture_type2.x))+
  stat_compare_means(aes(label = after_stat(p.signif)),
                     method = "t.test", ref.group = "BacterialControl",position = position_nudge(y = 0.03)) + 
  ggtitle("Stationary") +
  theme_bw() +
  scale_shape_manual("Microbiome", values = BactStat.sha)+
  scale_fill_manual("Phytoplankton",values = Cond.col) +
  My_Theme+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  scale_y_continuous(expand = expansion(mult = c(0.1, 0.2)))+
  scale_x_discrete(breaks=c('BacterialControl', 'CoccolithophoreOnly', 'CoccolithophoreBacteria' ,'DiatomOnly','DiatomBacteria'),
                     labels=c("Control","Coccolithophore","Coccolithophore + microbiome", "Diatom", "Diatom + microbiome"))+
  theme(
    plot.title=element_text(face='bold')
  )


Stationary_16S_dist_plot
ggsave("", width = 4, height = 4, units = "in", device = "pdf")



```


```{r distance comparisons Death}

Sven_phyto_rare_Death <- subset_samples(Sven_phyto_rare, Cultivation_Stage == "Death")
Sven_phyto_rare_Death

Totu_table = t(otu_table(Sven_phyto_rare_Death)) # transpose otu table
otu_table(Sven_phyto_rare_Death) = Totu_table

p = Sven_phyto_rare_Death
m = "bray"
s = "Original.names"
d = "Culture_type"

# Calculate distances

wu = vegan::vegdist(t(otu_table(p)), method = "bray")
wu.m = melt(as.matrix(wu))

colnames(wu.m) <- c("Var1", "Var2", "value")




# remove self-comparisons
wu.m = wu.m %>%
  filter(as.character(Var1) != as.character(Var2)) %>%
  mutate_if(is.factor, as.character)

# get sample data (S4 error OK and expected)
sd = data.frame(sample_data(p))%>%
  select(s, d)%>%
  mutate_if(is.factor, as.character) 

library(tibble)
sd <- tibble::rownames_to_column(sd, "Sample.names")
sd = subset(sd, select = -c(Original.names) )


# combined distances with sample data
colnames(sd) = c("Var1", "Culture_type")
wu.sd = left_join(wu.m, sd, by = "Var1")

colnames(sd) = c("Var2", "Culture_type2")
wu.sd = left_join(wu.sd, sd, by = "Var2")

wu.sd.sums = wu.sd %>%
  filter(Culture_type == 'BacterialControl') %>%
  mutate_if(is.factor,as.character)

# order factors
wu.sd.sums$Culture_type2 = factor(wu.sd.sums$Culture_type2, levels = c('BacterialControl', 'CoccolithophoreOnly', 'CoccolithophoreBacteria' ,'DiatomOnly','DiatomBacteria'))



sdpb = data.frame(sample_data(p))%>%
  select(s, d, Phyto,BacterialStatus)%>%
  mutate_if(is.factor, as.character)

sdpb <- tibble::rownames_to_column(sdpb, "Sample.names")
sdpb = subset(sdpb, select = -c(Original.names) )
colnames(sdpb) = c("Var2", "Culture_type2","Phyto", "BacterialStatus")
wu.sd.sums = left_join(wu.sd.sums, sdpb, by = "Var2")


wu.sd.sums$BacterialStatus = factor(wu.sd.sums$BacterialStatus,
                                           levels = c("None",
                                                      "Coccolithophore",
                                                      "Diatom"))

wu.sd.sums$Phyto<-as.factor(wu.sd.sums$Phyto)



Death_16S_dist_plot<-ggplot(wu.sd.sums, aes(x = Culture_type2.x, y = value,
                                                 fill = Phyto, 
                                                 group = Culture_type2.x))+
  geom_boxplot() +
  geom_point()+
  ylab("Bray distance") + xlab("") +
  stat_compare_means(method = "kruskal.test", size = 3,label.x = 3, label.y = .6, aes(group = Culture_type2.x))+
  stat_compare_means(aes(label = after_stat(p.signif)),
                     method = "t.test", ref.group = "BacterialControl") + 
  ggtitle("Death")+
  theme_bw()+
  scale_shape_manual("Microbiome", values = BactStat.sha)+
  scale_fill_manual("Phytoplankton",values = Cond.col) +
  ylab("Bray Distance")+
  My_Theme+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  scale_y_continuous(expand = expansion(mult = c(0.1, 0.2)))+
  scale_x_discrete(breaks=c('BacterialControl', 'CoccolithophoreOnly', 'CoccolithophoreBacteria' ,'DiatomOnly','DiatomBacteria'),
                     labels=c("Control","Coccolithophore","Coccolithophore + microbiome", "Diatom", "Diatom + microbiome"))+
  theme(
    plot.title=element_text(face='bold')
  )


Death_16S_dist_plot
ggsave("",width=4,height=4,units ="in", device="pdf")



```

```{r Combine distance plots}

Combined_dist_plot = ggpubr::ggarrange(Lag_16S_dist_plot,
                                       Exponential_16S_dist_plot,
                                       Stationary_16S_dist_plot,
                                       Death_16S_dist_plot,
                                       ncol = 2,
                                       nrow = 2,
                                       common.legend = T,
                                       legend = "right")


pdf("", width = 12, height = 11)
Combined_dist_plot
dev.off()
```

Stacked bar plot based on phylum (draft and clean version)

```{r stack bar phylum}

library(plyr)
library(scales)

# clean graph and cluster by Culture_type

library(dplyr)
library(microViz)
library(fantaxtic)
library(speedyseq)
library(phylosmith)

Sven_phyto_rare <- merge_treatments(Sven_phyto_rare, c("Culture_type", "Cultivation_Stage"))

SPR_clean <- name_na_taxa(Sven_phyto_rare, na_label = "Unidentified <tax> (<rank>)")

SPR_clean_test<-tax_fix(SPR_clean)

SPR_clean_test

SPR_clean_test %>%
  ps_arrange(
    Cultivation_Stage = factor(Cultivation_Stage, levels = c("Death","Stationary","Exponential", "Lag")))%>%
  ps_mutate(
    Culture_type = factor(Culture_type, levels = c("CoccolithophoreOnly","DiatomOnly", "BacterialControl","CoccolithophoreBacteria" ,"DiatomBacteria"))) %>% 
  comp_barplot("Phylum", n_taxa = 15, merge_other = FALSE, label = "Cultivation_Stage", sample_order = "asis") +
  facet_wrap(vars(Culture_type), scales = "free") + # scales = "free" is IMPORTANT!
  coord_flip() +
  theme(axis.ticks.y = element_blank(), strip.text = element_text(face = "bold"))
ggsave("",width=12,height=8,units ="in", device="pdf")


```




Stacked bar plot based on Order (clean version)

```{r stack bar Order}
SPR_clean_test %>%
  ps_arrange(
    Cultivation_Stage = factor(Cultivation_Stage, levels = c("Death","Stationary","Exponential", "Lag")))%>%
  ps_mutate(
    Culture_type = factor(Culture_type, levels = c("CoccolithophoreOnly","DiatomOnly", "BacterialControl","CoccolithophoreBacteria" ,"DiatomBacteria"))) %>% 
  comp_barplot("Order", n_taxa = 15, merge_other = FALSE, label = "Cultivation_Stage", sample_order = "asis") +
  facet_wrap(vars(Culture_type), scales = "free") + # scales = "free" is IMPORTANT!
  coord_flip() +
  theme(axis.ticks.y = element_blank(), strip.text = element_text(face = "bold"))
ggsave("",width=12,height=8,units ="in", device="pdf")

```

```{r Genus plot}

test<-SPR_clean_test %>%
  ps_arrange(
    Cultivation_Stage = factor(Cultivation_Stage, levels = c("Death","Stationary","Exponential", "Lag")))%>%
  ps_mutate(
    Culture_type = factor(Culture_type, levels = c("CoccolithophoreOnly","DiatomOnly", "BacterialControl","CoccolithophoreBacteria" ,"DiatomBacteria"))) %>% 
  comp_barplot("Genus", n_taxa = 15, merge_other = FALSE, label = "Cultivation_Stage", sample_order = "asis") +
  facet_wrap(vars(Culture_type), scales = "free") + # scales = "free" is IMPORTANT!
  coord_flip() +
  theme(axis.ticks.y = element_blank(), strip.text = element_text(face = "bold"))
ggsave("",width=12,height=8,units ="in", device="pdf")

test<-data.frame(test[["data"]])


 test<-as.data.frame(test)
```


```{r significantly affected taxa by stage Lag}

Control_Coccoonly_Lag <- subset_samples(Sven_phyto_rare_Lag, Culture_type %in% c("BacterialControl", "CoccolithophoreOnly"))

Control_Coccobact_Lag <- subset_samples(Sven_phyto_rare_Lag, Culture_type %in% c("BacterialControl", "CoccolithophoreBacteria"))

Control_DiatomOnly_Lag <- subset_samples(Sven_phyto_rare_Lag, Culture_type %in% c("BacterialControl", "DiatomOnly"))

Control_Diatombact_Lag <- subset_samples(Sven_phyto_rare_Lag, Culture_type %in% c("BacterialControl", "DiatomBacteria"))


#no hits after p adjusted so used just p value
Control_Coccoonly_Lag_df<-Control_Coccoonly_Lag %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) %>%
  psmelt() 

library(rstatix)
# do tests
KW_taxa_Control_Coccoonly_Lag<-Control_Coccoonly_Lag_df%>% 
  group_by(OTU) %>% 
  kruskal_test(Abundance ~ Culture_type) %>% 
  adjust_pvalue(method="fdr")%>% 
  filter(p < 0.05)

#no hits after p adjusted so used just p value
Control_Coccobact_Lag_df<-Control_Coccobact_Lag %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) %>%
  psmelt() 

# do tests
KW_taxa_Control_Coccobact_Lag<-Control_Coccobact_Lag_df%>% 
  group_by(OTU) %>% 
  kruskal_test(Abundance ~ Culture_type) %>% 
  adjust_pvalue(method="fdr")%>% 
  filter(p < 0.05)

#no hits after p adjusted so used just p value
Control_DiatomOnly_Lag_df<-Control_DiatomOnly_Lag %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) %>%
  psmelt() 

# do tests
KW_taxa_Control_DiatomOnly_Lag<-Control_DiatomOnly_Lag_df%>% 
  group_by(OTU) %>% 
  kruskal_test(Abundance ~ Culture_type) %>% 
  adjust_pvalue(method="fdr")%>% 
  filter(p < 0.05)

#no hits after p adjusted so used just p value
Control_Diatombact_Lag_df<-Control_Diatombact_Lag %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) %>%
  psmelt() 

# do tests
KW_taxa_Control_Diatombact_Lag<-Control_Diatombact_Lag_df%>% 
  group_by(OTU) %>% 
  kruskal_test(Abundance ~ Culture_type) %>% 
  adjust_pvalue(method="fdr")%>% 
  filter(p < 0.05)
```


```{r significantly affected taxa by stage Exponential}

Control_Coccoonly_Exponential <- subset_samples(Sven_phyto_rare_Exponential, Culture_type %in% c("BacterialControl", "CoccolithophoreOnly"))

Control_Coccobact_Exponential <- subset_samples(Sven_phyto_rare_Exponential, Culture_type %in% c("BacterialControl", "CoccolithophoreBacteria"))

Control_DiatomOnly_Exponential <- subset_samples(Sven_phyto_rare_Exponential, Culture_type %in% c("BacterialControl", "DiatomOnly"))

Control_Diatombact_Exponential <- subset_samples(Sven_phyto_rare_Exponential, Culture_type %in% c("BacterialControl", "DiatomBacteria"))


#no hits after p adjusted so used just p value
Control_Coccoonly_Exponential_df<-Control_Coccoonly_Exponential %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) %>%
  psmelt() 

library(rstatix)
# do tests
KW_taxa_Control_Coccoonly_Exponential<-Control_Coccoonly_Exponential_df%>% 
  group_by(OTU) %>% 
  kruskal_test(Abundance ~ Culture_type) %>% 
  adjust_pvalue(method="fdr")%>% 
  filter(p < 0.05)

#no hits after p adjusted so used just p value
Control_Coccobact_Exponential_df<-Control_Coccobact_Exponential %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) %>%
  psmelt() 

# do tests
KW_taxa_Control_Coccobact_Exponential<-Control_Coccobact_Exponential_df%>% 
  group_by(OTU) %>% 
  kruskal_test(Abundance ~ Culture_type) %>% 
  adjust_pvalue(method="fdr")%>% 
  filter(p < 0.05)

#no hits after p adjusted so used just p value
Control_DiatomOnly_Exponential_df<-Control_DiatomOnly_Exponential %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) %>%
  psmelt() 

# do tests
KW_taxa_Control_DiatomOnly_Exponential<-Control_DiatomOnly_Exponential_df%>% 
  group_by(OTU) %>% 
  kruskal_test(Abundance ~ Culture_type) %>% 
  adjust_pvalue(method="fdr")%>% 
  filter(p < 0.05)

#no hits after p adjusted so used just p value
Control_Diatombact_Exponential_df<-Control_Diatombact_Exponential %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) %>%
  psmelt() 

# do tests
KW_taxa_Control_Diatombact_Exponential<-Control_Diatombact_Exponential_df%>% 
  group_by(OTU) %>% 
  kruskal_test(Abundance ~ Culture_type) %>% 
  adjust_pvalue(method="fdr")%>% 
  filter(p < 0.05)
```

```{r significantly affected taxa by stage Stationary}

Control_Coccoonly_Stationary <- subset_samples(Sven_phyto_rare_Stationary, Culture_type %in% c("BacterialControl", "CoccolithophoreOnly"))

Control_Coccobact_Stationary <- subset_samples(Sven_phyto_rare_Stationary, Culture_type %in% c("BacterialControl", "CoccolithophoreBacteria"))

Control_DiatomOnly_Stationary <- subset_samples(Sven_phyto_rare_Stationary, Culture_type %in% c("BacterialControl", "DiatomOnly"))

Control_Diatombact_Stationary <- subset_samples(Sven_phyto_rare_Stationary, Culture_type %in% c("BacterialControl", "DiatomBacteria"))


#no hits after p adjusted so used just p value
Control_Coccoonly_Stationary_df<-Control_Coccoonly_Stationary %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) %>%
  psmelt() 

library(rstatix)
# do tests
KW_taxa_Control_Coccoonly_Stationary<-Control_Coccoonly_Stationary_df%>% 
  group_by(OTU) %>% 
  kruskal_test(Abundance ~ Culture_type) %>% 
  adjust_pvalue(method="fdr")%>% 
  filter(p < 0.05)

#no hits after p adjusted so used just p value
Control_Coccobact_Stationary_df<-Control_Coccobact_Stationary %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) %>%
  psmelt() 

# do tests
KW_taxa_Control_Coccobact_Stationary<-Control_Coccobact_Stationary_df%>% 
  group_by(OTU) %>% 
  kruskal_test(Abundance ~ Culture_type) %>% 
  adjust_pvalue(method="fdr")%>% 
  filter(p < 0.05)

#no hits after p adjusted so used just p value
Control_DiatomOnly_Stationary_df<-Control_DiatomOnly_Stationary %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) %>%
  psmelt() 

# do tests
KW_taxa_Control_DiatomOnly_Stationary<-Control_DiatomOnly_Stationary_df%>% 
  group_by(OTU) %>% 
  kruskal_test(Abundance ~ Culture_type) %>% 
  adjust_pvalue(method="fdr")%>% 
  filter(p < 0.05)

#no hits after p adjusted so used just p value
Control_Diatombact_Stationary_df<-Control_Diatombact_Stationary %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) %>%
  psmelt() 

# do tests
KW_taxa_Control_Diatombact_Stationary<-Control_Diatombact_Stationary_df%>% 
  group_by(OTU) %>% 
  kruskal_test(Abundance ~ Culture_type) %>% 
  adjust_pvalue(method="fdr")%>% 
  filter(p < 0.05)
```

```{r significantly affected taxa by stage Death}

Control_Coccoonly_Death <- subset_samples(Sven_phyto_rare_Death, Culture_type %in% c("BacterialControl", "CoccolithophoreOnly"))

Control_Coccobact_Death <- subset_samples(Sven_phyto_rare_Death, Culture_type %in% c("BacterialControl", "CoccolithophoreBacteria"))

Control_DiatomOnly_Death <- subset_samples(Sven_phyto_rare_Death, Culture_type %in% c("BacterialControl", "DiatomOnly"))

Control_Diatombact_Death <- subset_samples(Sven_phyto_rare_Death, Culture_type %in% c("BacterialControl", "DiatomBacteria"))


#no hits after p adjusted so used just p value
Control_Coccoonly_Death_df<-Control_Coccoonly_Death %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) %>%
  psmelt() 

library(rstatix)
# do tests
KW_taxa_Control_Coccoonly_Death<-Control_Coccoonly_Death_df%>% 
  group_by(OTU) %>% 
  kruskal_test(Abundance ~ Culture_type) %>% 
  adjust_pvalue(method="fdr")%>% 
  filter(p < 0.05)

#no hits after p adjusted so used just p value
Control_Coccobact_Death_df<-Control_Coccobact_Death %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) %>%
  psmelt() 

# do tests
KW_taxa_Control_Coccobact_Death<-Control_Coccobact_Death_df%>% 
  group_by(OTU) %>% 
  kruskal_test(Abundance ~ Culture_type) %>% 
  adjust_pvalue(method="fdr")%>% 
  filter(p < 0.05)

#no hits after p adjusted so used just p value
Control_DiatomOnly_Death_df<-Control_DiatomOnly_Death %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) %>%
  psmelt() 

# do tests
KW_taxa_Control_DiatomOnly_Death<-Control_DiatomOnly_Death_df%>% 
  group_by(OTU) %>% 
  kruskal_test(Abundance ~ Culture_type) %>% 
  adjust_pvalue(method="fdr")%>% 
  filter(p < 0.05)

#no hits after p adjusted so used just p value
Control_Diatombact_Death_df<-Control_Diatombact_Death %>% 
  transform_sample_counts(function(x) {x/sum(x)} ) %>%
  psmelt() 

# do tests
KW_taxa_Control_Diatombact_Death<-Control_Diatombact_Death_df%>% 
  group_by(OTU) %>% 
  kruskal_test(Abundance ~ Culture_type) %>% 
  adjust_pvalue(method="fdr")%>% 
  filter(p < 0.05)
```

```{r All significantly affected taxa plot Phylum}

#combine significant taxa across all tests
KW_taxa_All <- rbind(KW_taxa_Control_Coccoonly_Lag,KW_taxa_Control_Coccobact_Lag, KW_taxa_Control_DiatomOnly_Lag, KW_taxa_Control_Diatombact_Lag,KW_taxa_Control_Coccoonly_Exponential,KW_taxa_Control_Coccobact_Exponential, KW_taxa_Control_DiatomOnly_Exponential, KW_taxa_Control_Diatombact_Exponential,KW_taxa_Control_Coccoonly_Stationary,KW_taxa_Control_Coccobact_Stationary, KW_taxa_Control_DiatomOnly_Stationary, KW_taxa_Control_Diatombact_Stationary,KW_taxa_Control_Coccoonly_Death,KW_taxa_Control_Coccobact_Death, KW_taxa_Control_DiatomOnly_Death, KW_taxa_Control_Diatombact_Death,KW_taxa_Control_Coccoonly_Death,KW_taxa_Control_Coccobact_Death, KW_taxa_Control_DiatomOnly_Death, KW_taxa_Control_Diatombact_Death)

#remove redundant names and subset ps object
KW_taxa_All_names<-unique(KW_taxa_All$OTU)

KW_SPR_clean2 = prune_taxa(KW_taxa_All_names, SPR_clean) 

KW_SPR_clean2 <- name_na_taxa(KW_SPR_clean2, na_label = "Unidentified <tax> (<rank>)")

#KW_SPR_clean2<-tax_fix(KW_SPR_clean2)


KW_SPR_clean2 %>%
  ps_arrange(
    Cultivation_Stage = factor(Cultivation_Stage, levels = c("Death","Stationary","Exponential", "Lag")))%>%
  ps_mutate(
    Culture_type = factor(Culture_type, levels = c("CoccolithophoreOnly","DiatomOnly", "BacterialControl","CoccolithophoreBacteria" ,"DiatomBacteria"))) %>% 
  comp_barplot("Phylum", n_taxa = 15, merge_other = FALSE, label = "Cultivation_Stage", sample_order = "asis") +
  facet_wrap(vars(Culture_type), scales = "free") + # scales = "free" is IMPORTANT!
  coord_flip() +
  theme(axis.ticks.y = element_blank(), strip.text = element_text(face = "bold"))
ggsave("",width=12,height=8,units ="in", device="pdf")

```



```{r All significantly affected taxa plot Order}

KW_SPR_clean2 %>%
  ps_arrange(
    Cultivation_Stage = factor(Cultivation_Stage, levels = c("Death","Stationary","Exponential", "Lag")))%>%
  ps_mutate(
    Culture_type = factor(Culture_type, levels = c("CoccolithophoreOnly","DiatomOnly", "BacterialControl","CoccolithophoreBacteria" ,"DiatomBacteria"))) %>% 
  comp_barplot("Order", n_taxa = 15, merge_other = FALSE, label = "Cultivation_Stage", sample_order = "asis") +
  facet_wrap(vars(Culture_type), scales = "free") + # scales = "free" is IMPORTANT!
  coord_flip() +
  theme(axis.ticks.y = element_blank(), strip.text = element_text(face = "bold"))
ggsave("",width=12,height=8,units ="in", device="pdf")

```


```{r All significantly affected taxa plot Family}

KW_SPR_clean2 %>%
  ps_arrange(
    Cultivation_Stage = factor(Cultivation_Stage, levels = c("Death","Stationary","Exponential", "Lag")))%>%
  ps_mutate(
    Culture_type = factor(Culture_type, levels = c("CoccolithophoreOnly","DiatomOnly", "BacterialControl","CoccolithophoreBacteria" ,"DiatomBacteria"))) %>% 
  comp_barplot("Family", n_taxa = 15, merge_other = FALSE, label = "Cultivation_Stage", sample_order = "asis") +
  facet_wrap(vars(Culture_type), scales = "free") + # scales = "free" is IMPORTANT!
  coord_flip() +
  theme(axis.ticks.y = element_blank(), strip.text = element_text(face = "bold"))
ggsave("",width=12,height=8,units ="in", device="pdf")

```


```{r All significantly affected taxa plot Genus}

KW_SPR_clean2 %>%
  ps_arrange(
    Cultivation_Stage = factor(Cultivation_Stage, levels = c("Death","Stationary","Exponential", "Lag")))%>%
  ps_mutate(
    Culture_type = factor(Culture_type, levels = c("CoccolithophoreOnly","DiatomOnly", "BacterialControl","CoccolithophoreBacteria" ,"DiatomBacteria"))) %>% 
  comp_barplot("Genus", n_taxa = 15, merge_other = FALSE, label = "Cultivation_Stage", sample_order = "asis") +
  facet_wrap(vars(Culture_type), scales = "free") + # scales = "free" is IMPORTANT!
  coord_flip() +
  theme(axis.ticks.y = element_blank(), strip.text = element_text(face = "bold"))
ggsave("",width=12,height=8,units ="in", device="pdf")


```

```{r old taxa plot code}

SPR_clean2 <- SPR_clean %>%
  ps_select(Culture_type, Phyto, HoursFromStart, Cultivation_Stage, BacterialStatus,Culture_type_Cultivation_Stage) %>% # avoids lots of phyloseq::merge_samples warnings
  merge_samples2(group = "Culture_type_Cultivation_Stage", funs = list()) %>%
  tax_glom(taxrank = "Phylum")

SPR_clean2 %>%
  tax_filter(
    tax_level = "Phylum", min_prevalence = 0.001,
    prev_detection_threshold = 100
  ) %>%
  ps_mutate(
    Culture_type = factor(Culture_type, levels = c("CoccolithophoreOnly","DiatomOnly", "BacterialControl","CoccolithophoreBacteria" ,"DiatomBacteria"))) %>%  
  ps_mutate(
    Phyto = factor(Phyto, levels = c('Coccolithophore', 'Diatom', 'None'))) %>%  
  ps_mutate(
    Cultivation_Stage = factor(Cultivation_Stage, levels = c('Lag','Exponential','Stationary','Death'))) %>%  
  comp_barplot(
    tax_level = "Phylum", n_taxa = 10,
    bar_outline_colour = NA,
    bar_width = 0.7,
    taxon_renamer = toupper,
    facet_by = "Culture_type",
    label = "Cultivation_Stage",
    x = "Cultivation_Stage"
    ) + 
  theme(legend.text = element_text(size = 12),
        legend.title =element_text(size = 14, face="bold"),
        axis.title = element_text(size = 14), 
        axis.text.x = element_text(angle = 45, colour = "black", size = 10, vjust = 1, hjust = 1), 
        axis.text.y = element_text(colour = "black", size = 12),
        panel.background = element_rect(fill = "grey98"),
        panel.grid = element_line(color = "grey75"),
        panel.grid.major.y = element_blank(),
        panel.grid.major.x = element_blank())

ggsave("",width=12,height=8,units ="in", device="pdf")




# Clean graph and cluster by Culture_type

SPR_clean2_Order <- SPR_clean %>%
  ps_select(Culture_type, Phyto, HoursFromStart, Cultivation_Stage, BacterialStatus,Culture_type_Cultivation_Stage) %>% # avoids lots of phyloseq::merge_samples warnings
  merge_samples2(group = "Culture_type_Cultivation_Stage", funs = list()) %>%
  tax_glom(taxrank = "Order")

SPR_clean2_Order %>%
  tax_filter(
    tax_level = "Order", min_prevalence = 0.001,
    prev_detection_threshold = 100
  ) %>%
  ps_mutate(
    Culture_type = factor(Culture_type, levels = c("CoccolithophoreOnly","DiatomOnly", "BacterialControl","CoccolithophoreBacteria" ,"DiatomBacteria"))) %>%  
  ps_mutate(
    Phyto = factor(Phyto, levels = c('Coccolithophore', 'Diatom', 'None'))) %>%  
  ps_mutate(
    Cultivation_Stage = factor(Cultivation_Stage, levels = c('Lag','Exponential','Stationary','Death'))) %>%  
  comp_barplot(
    tax_level = "Order", n_taxa = 10,
    bar_outline_colour = NA,
    bar_width = 0.7,
    taxon_renamer = toupper,
    facet_by = "Culture_type",
    label = "Cultivation_Stage",
    x = "Cultivation_Stage"
  ) + 
  theme(legend.text = element_text(size = 12),
        legend.title =element_text(size = 14, face="bold"),
        axis.title = element_text(size = 14), 
        axis.text.x = element_text(angle = 45, colour = "black", size = 10, vjust = 1, hjust = 1), 
        axis.text.y = element_text(colour = "black", size = 12),
        panel.background = element_rect(fill = "grey98"),
        panel.grid = element_line(color = "grey75"),
        panel.grid.major.y = element_blank(),
        panel.grid.major.x = element_blank())

ggsave("",width=12,height=8,units ="in", device="pdf")
```

```{r unused Ternary plot Order}

library(microbiomeutilities)


tern_df<-prep_ternary(Sven_phyto_rare, group = "Phyto", abund.thres = 0.01, level = "Genus", prev.thres = 0.25)

library(ggtern)
# Replace empty with Other
tern_df$Phylum[tern_df$Phylum==""] <- "Other"

ggtern(data=tern_df, aes(x=None, y=Coccolithophore, z=Diatom)) + 
  geom_point(aes(color= Genus), 
             alpha=0.5, 
             show.legend=T, 
             size=3) +
  #scale_size(range=c(0, 6)) + 
  geom_mask() + 
  scale_colour_brewer(palette = "Paired") +
  theme_biome_utils()


#repeat with only significant ones
KW_tern_df<-prep_ternary(SPR_clean2, group = "Phyto", abund.thres = 0.01, level = "Genus", prev.thres = 0.1)

# Replace empty with Other
tern_df$Phylum[tern_df$Phylum==""] <- "Other"

ggtern(data=tern_df, aes(x=None, y=Coccolithophore, z=Diatom)) + 
  geom_point(aes(color= Order), 
             alpha=0.5, 
             show.legend=T, 
             size=3) +
  #scale_size(range=c(0, 6)) + 
  geom_mask() + 
  scale_colour_brewer(palette = "Paired") +
  theme_biome_utils()

```


```{r iCAMP analysis}
#Create distance plots
library(dysbiosisR)
Sven_phyto %>%
  refseq() %>%
  Biostrings::writeXStringSet("C:/Users/bogda/Documents/PhD/sven/asvs.fasta", append=FALSE,
                              compress=FALSE, compression_level=NA, format="fasta")


physeq<-Sven_phyto_rare
stage="Exponential"
Phyto="Diatom"
dist2control<-function(stage,physeq,Phyto){
  samples2keep<-sample_names(physeq)[physeq@sam_data$Cultivation_Stage==stage]
  samples2keep_treat<-sample_names(physeq)[physeq@sam_data$Phyto %in% c("None",Phyto)]
  samples2keep<-intersect(samples2keep,samples2keep_treat)
  tempphy<-prune_samples(samples2keep,physeq)
  dist.mat <- phyloseq::distance(tempphy, "bray")
  dysbiosis_2 <- euclideanDistCentroids(tempphy,
                                        dist_mat = dist.mat,
                                        use_squared = TRUE,
                                        group_col = "BacterialStatus",
                                        control_label = "Present",
                                        case_label = "Absent")
  dysbiosis_2
}
dist2control(stage="Lag",physeq=Sven_phyto_rare,Phyto="Coccolithophore")

library(iCAMP)
tree_sven<-read_tree("")

phyloseq::phy_tree(Sven_phyto_rare)<-tree_sven
Sven_phyto_rare_dia_bac<-prune_samples(sample_names(Sven_phyto_rare)[Sven_phyto_rare@sam_data$Phyto=="Diatom" &
                                                                       Sven_phyto_rare@sam_data$BacterialStatus=="Present"], Sven_phyto_rare)
Sven_phyto_rare_dia_nobac<-prune_samples(sample_names(Sven_phyto_rare)[Sven_phyto_rare@sam_data$Phyto=="Diatom" &
                                                                         Sven_phyto_rare@sam_data$BacterialStatus=="Absent"], Sven_phyto_rare)
Sven_phyto_rare_coc_bac<-prune_samples(sample_names(Sven_phyto_rare)[Sven_phyto_rare@sam_data$Phyto=="Coccolithophore" &
                                                                       Sven_phyto_rare@sam_data$BacterialStatus=="Present"], Sven_phyto_rare)
Sven_phyto_rare_coc_nobac<-prune_samples(sample_names(Sven_phyto_rare)[Sven_phyto_rare@sam_data$Phyto=="Coccolithophore" &
                                                                         Sven_phyto_rare@sam_data$BacterialStatus=="Absent"], Sven_phyto_rare)

Sven_phyto_rare_bac<-prune_samples(sample_names(Sven_phyto_rare)[Sven_phyto_rare@sam_data$Phyto=="None" &
                                                                   Sven_phyto_rare@sam_data$BacterialStatus=="Present"], Sven_phyto_rare)

#coc_nobac analysis
unlink("iCAMP.iCAMP.Confidence.detail.rda")
unlink("iCAMP.process.CbMPDiCBraya.csv")
unlink("path.rda")
unlink("pd.bin")
unlink("pd.desc")
unlink("pd.taxon.name.csv")
physeq<-Sven_phyto_rare_coc_nobac
physeq<-prune_taxa(names(taxa_sums(physeq)[taxa_sums(physeq)>1]),physeq)
comm=physeq@otu_table@.Data
tree=physeq@phy_tree
meta.group=data.frame(meta.com=physeq@sam_data$Cultivation_Stage)
rownames(meta.group)=rownames(comm)
wd0=getwd() # please change to the folder you want to save the pd.big output.
save.wd="./"
nworker=1 # parallel computing thread number
rand.time=1000 # usually use 1000 for real data.
bin.size.limit=50 # for real data, usually use a proper number
Sven_phyto_rare_coc_nobac_icamp=iCAMP::icamp.cm(comm=comm, tree=tree, meta.group=meta.group,
                                                pd.wd="./", rand=rand.time, nworker=nworker,
                                                bin.size.limit=bin.size.limit,phylo.metric="bMNTD")
Sven_phyto_rare_coc_nobac_icamp$CbMNTDiCBraya$S1_Stage<-Sven_phyto_rare_coc_nobac@sam_data$Cultivation_Stage[match(Sven_phyto_rare_coc_nobac_icamp$CbMNTDiCBraya$sample1,sample_names(Sven_phyto_rare_coc_nobac))]
Sven_phyto_rare_coc_nobac_icamp$CbMNTDiCBraya$S2_Stage<-Sven_phyto_rare_coc_nobac@sam_data$Cultivation_Stage[match(Sven_phyto_rare_coc_nobac_icamp$CbMNTDiCBraya$sample2,sample_names(Sven_phyto_rare_coc_nobac))]
Sven_phyto_rare_coc_nobac_icamp_x<-Sven_phyto_rare_coc_nobac_icamp$CbMNTDiCBraya
Sven_phyto_rare_coc_nobac_icamp_x<-Sven_phyto_rare_coc_nobac_icamp_x[Sven_phyto_rare_coc_nobac_icamp_x$S1_Stage==Sven_phyto_rare_coc_nobac_icamp_x$S2_Stage,]
Sven_phyto_rare_coc_nobac_icamp_x<-reshape2::melt(Sven_phyto_rare_coc_nobac_icamp_x)
Sven_phyto_rare_coc_nobac_icamp_x$S2_Stage<-NULL
Sven_phyto_rare_coc_nobac_icamp_x<-aggregate(x=Sven_phyto_rare_coc_nobac_icamp_x$value,
                                             by=list(Sven_phyto_rare_coc_nobac_icamp_x$S1_Stage,Sven_phyto_rare_coc_nobac_icamp_x$variable),
                                             FUN=mean)
Sven_phyto_rare_coc_nobac_icamp_x$CustOrd<-factor(Sven_phyto_rare_coc_nobac_icamp_x$Group.1,levels=c("Lag","Exponential","Stationary","Death"))
Sven_phyto_rare_coc_nobac_icamp_x$Group.2<-gsub(".", " ", Sven_phyto_rare_coc_nobac_icamp_x$Group.2,fixed = TRUE)
Sven_phyto_rare_coc_nobac_icamp_x$Group.2 = factor(Sven_phyto_rare_coc_nobac_icamp_x$Group.2, levels = c('Heterogeneous Selection','Homogeneous Selection', 'Dispersal Limitation', 'Homogenizing Dispersal', 'Drift and Others'))

iCAMP.col <- c("Heterogeneous Selection" = "#332288",
               "Homogeneous Selection" = "#23C3E4",
               "Dispersal Limitation" = "red",
               "Homogenizing Dispersal" = "#F25F14FF",
               "Drift and Others" = "#EBD239")



ggplot(Sven_phyto_rare_coc_nobac_icamp_x) +
  aes(x = CustOrd, y = x, fill = Group.2) +
  geom_col() +
  scale_fill_manual("Ecological Process",values = iCAMP.col)+
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(angle=45, colour = "black", vjust=1, hjust = 0.9, size=12),
        axis.text.y = element_text(colour = "black", size=12),
        axis.title.y = element_text(face="bold",size=12),
        legend.title =element_text(size =12, face="bold"),
        legend.text = element_text(size = 12),
        legend.position="right",
        legend.key.size = unit(0.5, "cm"),
        #strip.text.x = element_text(size=20, face="bold"),
        strip.text.y = element_text(size=12, face="bold"),
        panel.background = element_blank(),
        panel.border = element_rect(fill = NA, colour = "black"),                                                                    
        strip.background = element_rect(colour="black"))+
  labs(y="Relative contribution")


#coc_bac analysis

unlink("iCAMP.iCAMP.Confidence.detail.rda")
unlink("iCAMP.process.CbMPDiCBraya.csv")
unlink("path.rda")
unlink("pd.bin")
unlink("pd.desc")
unlink("pd.taxon.name.csv")
physeq<-Sven_phyto_rare_coc_bac
physeq<-prune_taxa(names(taxa_sums(physeq)[taxa_sums(physeq)>1]),physeq)
comm=physeq@otu_table@.Data
tree=physeq@phy_tree
meta.group=data.frame(meta.com=physeq@sam_data$Cultivation_Stage)
rownames(meta.group)=rownames(comm)
wd0=getwd() # please change to the folder you want to save the pd.big output.
save.wd="./"
nworker=1 # parallel computing thread number
rand.time=1000 # usually use 1000 for real data.
bin.size.limit=50 # for real data, usually use a proper number
Sven_phyto_rare_coc_bac_icamp=icamp.cm(comm=comm, tree=tree, meta.group=meta.group,
                                       pd.wd="./", rand=rand.time, nworker=nworker,
                                       bin.size.limit=bin.size.limit,phylo.metric="bMNTD")
Sven_phyto_rare_coc_bac_icamp$CbMNTDiCBraya$S1_Stage<-Sven_phyto_rare_coc_bac@sam_data$Cultivation_Stage[match(Sven_phyto_rare_coc_bac_icamp$CbMNTDiCBraya$sample1,sample_names(Sven_phyto_rare_coc_bac))]
Sven_phyto_rare_coc_bac_icamp$CbMNTDiCBraya$S2_Stage<-Sven_phyto_rare_coc_bac@sam_data$Cultivation_Stage[match(Sven_phyto_rare_coc_bac_icamp$CbMNTDiCBraya$sample2,sample_names(Sven_phyto_rare_coc_bac))]
Sven_phyto_rare_coc_bac_icamp_x<-Sven_phyto_rare_coc_bac_icamp$CbMNTDiCBraya
Sven_phyto_rare_coc_bac_icamp_x<-Sven_phyto_rare_coc_bac_icamp_x[Sven_phyto_rare_coc_bac_icamp_x$S1_Stage==Sven_phyto_rare_coc_bac_icamp_x$S2_Stage,]
Sven_phyto_rare_coc_bac_icamp_x<-reshape2::melt(Sven_phyto_rare_coc_bac_icamp_x)
Sven_phyto_rare_coc_bac_icamp_x$S2_Stage<-NULL
Sven_phyto_rare_coc_bac_icamp_x<-aggregate(x=Sven_phyto_rare_coc_bac_icamp_x$value,
                                           by=list(Sven_phyto_rare_coc_bac_icamp_x$S1_Stage,Sven_phyto_rare_coc_bac_icamp_x$variable),
                                           FUN=mean)
Sven_phyto_rare_coc_bac_icamp_x$CustOrd<-factor(Sven_phyto_rare_coc_bac_icamp_x$Group.1,levels=c("Lag","Exponential","Stationary","Death"))

Sven_phyto_rare_coc_bac_icamp_x$Group.2<-gsub(".", " ", Sven_phyto_rare_coc_bac_icamp_x$Group.2,fixed = TRUE)
Sven_phyto_rare_coc_bac_icamp_x$Group.2 = factor(Sven_phyto_rare_coc_bac_icamp_x$Group.2, levels = c('Heterogeneous Selection','Homogeneous Selection', 'Dispersal Limitation', 'Homogenizing Dispersal', 'Drift and Others'))



ggplot(Sven_phyto_rare_coc_bac_icamp_x) +
  aes(x = CustOrd, y = x, fill = Group.2) +
  geom_col() +
  scale_fill_manual("Ecological Process",values = iCAMP.col)+
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(angle=45, colour = "black", vjust=1, hjust = 0.9, size=12),
        axis.text.y = element_text(colour = "black", size=12),
        axis.title.y = element_text(face="bold",size=12),
        legend.title =element_text(size =12, face="bold"),
        legend.text = element_text(size = 12),
        legend.position="right",
        legend.key.size = unit(0.5, "cm"),
        #strip.text.x = element_text(size=20, face="bold"),
        strip.text.y = element_text(size=12, face="bold"),
        panel.background = element_blank(),
        panel.border = element_rect(fill = NA, colour = "black"),                                                                    
        strip.background = element_rect(colour="black"))+
  labs(y="Relative contribution")


#dia_bac

unlink("iCAMP.iCAMP.Confidence.detail.rda")
unlink("iCAMP.process.CbMPDiCBraya.csv")
unlink("path.rda")
unlink("pd.bin")
unlink("pd.desc")
unlink("pd.taxon.name.csv")
physeq<-Sven_phyto_rare_dia_bac
physeq<-prune_taxa(names(taxa_sums(physeq)[taxa_sums(physeq)>1]),physeq)
comm=physeq@otu_table@.Data
tree=physeq@phy_tree
meta.group=data.frame(meta.com=physeq@sam_data$Cultivation_Stage)
rownames(meta.group)=rownames(comm)
wd0=getwd() # please change to the folder you want to save the pd.big output.
save.wd="./"
nworker=1 # parallel computing thread number
rand.time=1000 # usually use 1000 for real data.
bin.size.limit=50 # for real data, usually use a proper number
Sven_phyto_rare_dia_bac_icamp=icamp.cm(comm=comm, tree=tree, meta.group=meta.group,
                                       pd.wd="./", rand=rand.time, nworker=nworker,
                                       bin.size.limit=bin.size.limit,phylo.metric="bMNTD")
Sven_phyto_rare_dia_bac_icamp$CbMNTDiCBraya$S1_Stage<-Sven_phyto_rare_dia_bac@sam_data$Cultivation_Stage[match(Sven_phyto_rare_dia_bac_icamp$CbMNTDiCBraya$sample1,sample_names(Sven_phyto_rare_dia_bac))]
Sven_phyto_rare_dia_bac_icamp$CbMNTDiCBraya$S2_Stage<-Sven_phyto_rare_dia_bac@sam_data$Cultivation_Stage[match(Sven_phyto_rare_dia_bac_icamp$CbMNTDiCBraya$sample2,sample_names(Sven_phyto_rare_dia_bac))]
Sven_phyto_rare_dia_bac_icamp_x<-Sven_phyto_rare_dia_bac_icamp$CbMNTDiCBraya
Sven_phyto_rare_dia_bac_icamp_x<-Sven_phyto_rare_dia_bac_icamp_x[Sven_phyto_rare_dia_bac_icamp_x$S1_Stage==Sven_phyto_rare_dia_bac_icamp_x$S2_Stage,]
Sven_phyto_rare_dia_bac_icamp_x<-reshape2::melt(Sven_phyto_rare_dia_bac_icamp_x)
Sven_phyto_rare_dia_bac_icamp_x$S2_Stage<-NULL
Sven_phyto_rare_dia_bac_icamp_x<-aggregate(x=Sven_phyto_rare_dia_bac_icamp_x$value,
                                           by=list(Sven_phyto_rare_dia_bac_icamp_x$S1_Stage,Sven_phyto_rare_dia_bac_icamp_x$variable),
                                           FUN=mean)
Sven_phyto_rare_dia_bac_icamp_x$CustOrd<-factor(Sven_phyto_rare_dia_bac_icamp_x$Group.1,levels=c("Lag","Exponential","Stationary","Death"))


Sven_phyto_rare_dia_bac_icamp_x$Group.2<-gsub(".", " ", Sven_phyto_rare_dia_bac_icamp_x$Group.2,fixed = TRUE)
Sven_phyto_rare_dia_bac_icamp_x$Group.2 = factor(Sven_phyto_rare_dia_bac_icamp_x$Group.2, levels = c('Heterogeneous Selection','Homogeneous Selection', 'Dispersal Limitation', 'Homogenizing Dispersal', 'Drift and Others'))



ggplot(Sven_phyto_rare_dia_bac_icamp_x) +
  aes(x = CustOrd, y = x, fill = Group.2) +
  geom_col() +
  scale_fill_manual("Ecological Process",values = iCAMP.col)+
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(angle=45, colour = "black", vjust=1, hjust = 0.9, size=12),
        axis.text.y = element_text(colour = "black", size=12),
        axis.title.y = element_text(face="bold",size=12),
        legend.title =element_text(size =12, face="bold"),
        legend.text = element_text(size = 12),
        legend.position="right",
        legend.key.size = unit(0.5, "cm"),
        #strip.text.x = element_text(size=20, face="bold"),
        strip.text.y = element_text(size=12, face="bold"),
        panel.background = element_blank(),
        panel.border = element_rect(fill = NA, colour = "black"),                                                                    
        strip.background = element_rect(colour="black"))+
  labs(y="Relative contribution")


#dia_nobac

unlink("iCAMP.iCAMP.Confidence.detail.rda")
unlink("iCAMP.process.CbMPDiCBraya.csv")
unlink("path.rda")
unlink("pd.bin")
unlink("pd.desc")
unlink("pd.taxon.name.csv")
physeq<-Sven_phyto_rare_dia_nobac
physeq<-prune_taxa(names(taxa_sums(physeq)[taxa_sums(physeq)>1]),physeq)
comm=physeq@otu_table@.Data
tree=physeq@phy_tree
meta.group=data.frame(meta.com=physeq@sam_data$Cultivation_Stage)
rownames(meta.group)=rownames(comm)
wd0=getwd() # please change to the folder you want to save the pd.big output.
save.wd="./"
nworker=1 # parallel computing thread number
rand.time=1000 # usually use 1000 for real data.
bin.size.limit=50 # for real data, usually use a proper number
Sven_phyto_rare_dia_nobac_icamp=icamp.cm(comm=comm, tree=tree, meta.group=meta.group,
                                         pd.wd="./", rand=rand.time, nworker=nworker,
                                         bin.size.limit=bin.size.limit,phylo.metric="bMNTD")
Sven_phyto_rare_dia_nobac_icamp$CbMNTDiCBraya$S1_Stage<-Sven_phyto_rare_dia_nobac@sam_data$Cultivation_Stage[match(Sven_phyto_rare_dia_nobac_icamp$CbMNTDiCBraya$sample1,sample_names(Sven_phyto_rare_dia_nobac))]
Sven_phyto_rare_dia_nobac_icamp$CbMNTDiCBraya$S2_Stage<-Sven_phyto_rare_dia_nobac@sam_data$Cultivation_Stage[match(Sven_phyto_rare_dia_nobac_icamp$CbMNTDiCBraya$sample2,sample_names(Sven_phyto_rare_dia_nobac))]
Sven_phyto_rare_dia_nobac_icamp_x<-Sven_phyto_rare_dia_nobac_icamp$CbMNTDiCBraya
Sven_phyto_rare_dia_nobac_icamp_x<-Sven_phyto_rare_dia_nobac_icamp_x[Sven_phyto_rare_dia_nobac_icamp_x$S1_Stage==Sven_phyto_rare_dia_nobac_icamp_x$S2_Stage,]
Sven_phyto_rare_dia_nobac_icamp_x<-reshape2::melt(Sven_phyto_rare_dia_nobac_icamp_x)
Sven_phyto_rare_dia_nobac_icamp_x$S2_Stage<-NULL
Sven_phyto_rare_dia_nobac_icamp_x<-aggregate(x=Sven_phyto_rare_dia_nobac_icamp_x$value,
                                             by=list(Sven_phyto_rare_dia_nobac_icamp_x$S1_Stage,Sven_phyto_rare_dia_nobac_icamp_x$variable),
                                             FUN=mean)
Sven_phyto_rare_dia_nobac_icamp_x$CustOrd<-factor(Sven_phyto_rare_dia_nobac_icamp_x$Group.1,levels=c("Lag","Exponential","Stationary","Death"))



Sven_phyto_rare_dia_nobac_icamp_x$Group.2<-gsub(".", " ", Sven_phyto_rare_dia_nobac_icamp_x$Group.2,fixed = TRUE)
Sven_phyto_rare_dia_nobac_icamp_x$Group.2 = factor(Sven_phyto_rare_dia_nobac_icamp_x$Group.2, levels = c('Heterogeneous Selection','Homogeneous Selection', 'Dispersal Limitation', 'Homogenizing Dispersal', 'Drift and Others'))



ggplot(Sven_phyto_rare_dia_nobac_icamp_x) +
  aes(x = CustOrd, y = x, fill = Group.2) +
  geom_col() +
  scale_fill_manual("Ecological Process",values = iCAMP.col)+
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(angle=45, colour = "black", vjust=1, hjust = 0.9, size=12),
        axis.text.y = element_text(colour = "black", size=12),
        axis.title.y = element_text(face="bold",size=12),
        legend.title =element_text(size =12, face="bold"),
        legend.text = element_text(size = 12),
        legend.position="right",
        legend.key.size = unit(0.5, "cm"),
        #strip.text.x = element_text(size=20, face="bold"),
        strip.text.y = element_text(size=12, face="bold"),
        panel.background = element_blank(),
        panel.border = element_rect(fill = NA, colour = "black"),                                                                    
        strip.background = element_rect(colour="black"))+
  labs(y="Relative contribution")

#bac


unlink("iCAMP.iCAMP.Confidence.detail.rda")
unlink("iCAMP.process.CbMPDiCBraya.csv")
unlink("path.rda")
unlink("pd.bin")
unlink("pd.desc")
unlink("pd.taxon.name.csv")
physeq<-Sven_phyto_rare_bac
physeq<-prune_taxa(names(taxa_sums(physeq)[taxa_sums(physeq)>1]),physeq)
comm=physeq@otu_table@.Data
tree=physeq@phy_tree
meta.group=data.frame(meta.com=physeq@sam_data$Cultivation_Stage)
rownames(meta.group)=rownames(comm)
wd0=getwd() # please change to the folder you want to save the pd.big output.
save.wd="./"
nworker=1 # parallel computing thread number
rand.time=1000 # usually use 1000 for real data.
bin.size.limit=50 # for real data, usually use a proper number
Sven_phyto_rare_bac_icamp=icamp.cm(comm=comm, tree=tree, meta.group=meta.group,
                                   pd.wd="./", rand=rand.time, nworker=nworker,
                                   bin.size.limit=bin.size.limit,phylo.metric="bMNTD")
Sven_phyto_rare_bac_icamp$CbMNTDiCBraya$S1_Stage<-Sven_phyto_rare_bac@sam_data$Cultivation_Stage[match(Sven_phyto_rare_bac_icamp$CbMNTDiCBraya$sample1,sample_names(Sven_phyto_rare_bac))]
Sven_phyto_rare_bac_icamp$CbMNTDiCBraya$S2_Stage<-Sven_phyto_rare_bac@sam_data$Cultivation_Stage[match(Sven_phyto_rare_bac_icamp$CbMNTDiCBraya$sample2,sample_names(Sven_phyto_rare_bac))]
Sven_phyto_rare_bac_icamp_x<-Sven_phyto_rare_bac_icamp$CbMNTDiCBraya
Sven_phyto_rare_bac_icamp_x<-Sven_phyto_rare_bac_icamp_x[Sven_phyto_rare_bac_icamp_x$S1_Stage==Sven_phyto_rare_bac_icamp_x$S2_Stage,]
Sven_phyto_rare_bac_icamp_x<-reshape2::melt(Sven_phyto_rare_bac_icamp_x)
Sven_phyto_rare_bac_icamp_x$S2_Stage<-NULL
Sven_phyto_rare_bac_icamp_x<-aggregate(x=Sven_phyto_rare_bac_icamp_x$value,
                                       by=list(Sven_phyto_rare_bac_icamp_x$S1_Stage,Sven_phyto_rare_bac_icamp_x$variable),
                                       FUN=mean)
Sven_phyto_rare_bac_icamp_x$CustOrd<-factor(Sven_phyto_rare_bac_icamp_x$Group.1,levels=c("Lag","Exponential","Stationary","Death"))


Sven_phyto_rare_bac_icamp_x$Group.2<-gsub(".", " ", Sven_phyto_rare_bac_icamp_x$Group.2,fixed = TRUE)
Sven_phyto_rare_bac_icamp_x$Group.2 = factor(Sven_phyto_rare_bac_icamp_x$Group.2, levels = c('Heterogeneous Selection','Homogeneous Selection', 'Dispersal Limitation', 'Homogenizing Dispersal', 'Drift and Others'))



ggplot(Sven_phyto_rare_bac_icamp_x) +
  aes(x = CustOrd, y = x, fill = Group.2) +
  geom_col() +
  scale_fill_manual("Ecological Process",values = iCAMP.col)+
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(angle=45, colour = "black", vjust=1, hjust = 0.9, size=12),
        axis.text.y = element_text(colour = "black", size=12),
        axis.title.y = element_text(face="bold",size=12),
        legend.title =element_text(size =12, face="bold"),
        legend.text = element_text(size = 12),
        legend.position="right",
        legend.key.size = unit(0.5, "cm"),
        #strip.text.x = element_text(size=20, face="bold"),
        strip.text.y = element_text(size=12, face="bold"),
        panel.background = element_blank(),
        panel.border = element_rect(fill = NA, colour = "black"),                                                                    
        strip.background = element_rect(colour="black"))+
  labs(y="Relative contribution")


Sven_phyto_rare_dia_nobac_icamp_x$Group="Diatom Only"
Sven_phyto_rare_dia_bac_icamp_x$Group="Diatom + Bacteria"
Sven_phyto_rare_coc_nobac_icamp_x$Group="Coccolithophore Only"
Sven_phyto_rare_coc_bac_icamp_x$Group="Coccolithophore + Bacteria"
Sven_phyto_rare_bac_icamp_x$Group="Bacteria Only"
Sven_phyto_rare_dia_nobac_icamp_x$
  Sven_phyto_rare_coc_nobac_icamp_x
sel_icamp_res<-rbind(Sven_phyto_rare_bac_icamp_x,Sven_phyto_rare_coc_bac_icamp_x,
                     Sven_phyto_rare_coc_nobac_icamp_x,Sven_phyto_rare_dia_bac_icamp_x,
                     Sven_phyto_rare_dia_nobac_icamp_x)
sel_icamp_res$CustOrd<-factor(sel_icamp_res$Group.1,levels=c("Lag","Exponential","Stationary","Death"))
sel_icamp_res$Group<-factor(sel_icamp_res$Group,levels=c("Coccolithophore Only","Diatom Only", "Bacteria Only","Coccolithophore + Bacteria" ,"Diatom + Bacteria"))


ggplot(sel_icamp_res) +
  aes(x = CustOrd, y = x, fill = Group.2) +
  geom_col() +
  scale_fill_manual("Ecological Process",values = iCAMP.col)+
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(angle=45, colour = "black", vjust=1, hjust = 0.9, size=13),
        axis.text.y = element_text(colour = "black", size=13),
        axis.title.y = element_text(face="bold",size=13),
        legend.title =element_text(size =13, face="bold"),
        legend.text = element_text(size = 13),
        legend.position = c(1, 0), legend.justification = c(1, 0),
        legend.key.size = unit(0.7, "cm"),
        strip.text.x = element_text(size=13, face="bold"),
        strip.text.y = element_text(size=13, face="bold"),
        panel.background = element_blank(),
        panel.border = element_rect(fill = NA, colour = "black"),                                                                    
        strip.background = element_rect(colour="black"))+
  labs(y="Relative contribution") +
  facet_wrap(vars(Group))

ggsave("",width=12,height=8,units ="in", device="pdf")

```


```{r core ternary analysis}

RMean = 0.01
coremethod="elbow"




create_core_comm_sven<-function(physeq,nReads){
  library(tidyverse)
  library(reshape2)
  library(vegan)
  library(ggsci)
  library(phyloseq)
  theme_set(theme_light())
  nReads=nReads
  otu<-t(physeq@otu_table@.Data)
  otu_PA <- 1*((otu>0)==1)                                               # presence-absence data
  otu_occ <- rowSums(otu_PA)/ncol(otu_PA)                                # occupancy calculation
  otu_rel <- apply(decostand(otu, method="total", MARGIN=2),1, mean)     # relative abundance  
  occ_abun <- data.frame(otu_occ=otu_occ, otu_rel=otu_rel) %>%           # combining occupancy and abundance data frame
    rownames_to_column('otu')
  map<-sample_data(physeq)
  
  PresenceSum <- data.frame(otu = as.factor(row.names(otu)), otu) %>% 
    gather(Sample, abun, -otu) %>%
    dplyr::left_join(map, by = 'Sample') %>%
    group_by(otu, Location) %>%
    summarise(plot_freq=sum(abun>0)/length(abun),        # frequency of detection between time points
              coreSite=ifelse(plot_freq == 1, 1, 0), # 1 only if occupancy 1 with specific genotype, 0 if not
              detect=ifelse(plot_freq > 0, 1, 0)) %>%    # 1 if detected and 0 if not detected with specific genotype
    group_by(otu) %>%
    dplyr::summarise(sumF=sum(plot_freq),
                     sumG=sum(coreSite),
                     nS=length(Location)*2,
                     Index=(sumF+sumG)/nS) # calculating weighting Index based on number of time points detected and 
  
  otu_ranked <- occ_abun %>%
    dplyr::left_join(PresenceSum, by='otu') %>%
    transmute(otu=otu,
              rank=Index) %>%
    arrange(desc(rank))
  
  BCaddition <- NULL
  
  otu_start=otu_ranked$otu[1]
  start_matrix <- as.matrix(otu[otu_start,])
  start_matrix <- t(start_matrix)
  x <- apply(combn(ncol(start_matrix), 2), 2, function(x) sum(abs(start_matrix[,x[1]]- start_matrix[,x[2]]))/(2*nReads))
  x_names <- apply(combn(ncol(start_matrix), 2), 2, function(x) paste(colnames(start_matrix)[x], collapse=' - '))
  df_s <- data.frame(x_names,x)
  names(df_s)[2] <- 1 
  BCaddition <- rbind(BCaddition,df_s)
  
  for(i in 2:200){
    otu_add=otu_ranked$otu[i]
    add_matrix <- as.matrix(otu[otu_add,])
    add_matrix <- t(add_matrix)
    start_matrix <- rbind(start_matrix, add_matrix)
    x <- apply(combn(ncol(start_matrix), 2), 2, function(x) sum(abs(start_matrix[,x[1]]-start_matrix[,x[2]]))/(2*nReads))
    x_names <- apply(combn(ncol(start_matrix), 2), 2, function(x) paste(colnames(start_matrix)[x], collapse=' - '))
    df_a <- data.frame(x_names,x)
    names(df_a)[2] <- i 
    BCaddition <- dplyr::left_join(BCaddition, df_a, by=c('x_names'))  
  }
  
  x <-  apply(combn(ncol(otu), 2), 2, function(x) sum(abs(otu[,x[1]]-otu[,x[2]]))/(2*nReads))
  x_names <- apply(combn(ncol(otu), 2), 2, function(x) paste(colnames(otu)[x], collapse=' - '))
  df_full <- data.frame(x_names,x)
  names(df_full)[2] <- length(rownames(otu))
  BCfull <- dplyr::left_join(BCaddition,df_full, by='x_names')
  
  rownames(BCfull) <- BCfull$x_names
  temp_BC <- BCfull
  temp_BC$x_names <- NULL
  temp_BC_matrix <- as.matrix(temp_BC)
  
  BC_ranked <- data.frame(rank = as.factor(row.names(t(temp_BC_matrix))),t(temp_BC_matrix)) %>% 
    gather(comparison, BC, -rank) %>%
    group_by(rank) %>%
    dplyr::summarise(MeanBC=mean(BC)) %>%
    arrange(-desc(MeanBC)) %>%
    mutate(proportionBC=MeanBC/max(MeanBC))
  Increase=BC_ranked$MeanBC[-1]/BC_ranked$MeanBC[-length(BC_ranked$MeanBC)]
  increaseDF <- data.frame(IncreaseBC=c(0,(Increase)), rank=factor(c(1:(length(Increase)+1))))
  BC_ranked <- dplyr::left_join(BC_ranked, increaseDF)
  BC_ranked <- BC_ranked[-nrow(BC_ranked),]
  
  fo_difference <- function(pos){
    left <- (BC_ranked[pos, 2] - BC_ranked[1, 2]) / pos
    right <- (BC_ranked[nrow(BC_ranked), 2] - BC_ranked[pos, 2]) / (nrow(BC_ranked) - pos)
    return(left - right)
  }
  BC_ranked$fo_diffs <- sapply(1:nrow(BC_ranked), fo_difference)
  elbow <- which.max(BC_ranked$fo_diffs)
  
  lastCall <- last(as.numeric(as.character(BC_ranked$rank[(BC_ranked$IncreaseBC>=1.02)])))
  
  graph<-ggplot(BC_ranked[1:200,], aes(x=factor(BC_ranked$rank[1:200], levels=BC_ranked$rank[1:200]))) +
    geom_point(aes(y=proportionBC)) +
    theme_classic() + theme(strip.background = element_blank(),axis.text.x = element_text(size=7, angle=45)) +
    geom_vline(xintercept=elbow, lty=3, col='red', cex=.5) +
    geom_vline(xintercept=lastCall, lty=3, col='blue', cex=.5) +
    labs(x='ranked OTUs',y='Bray-Curtis similarity') +
    annotate(geom="text", x=elbow+10, y=.15, label=paste("Elbow method"," (",elbow,")", sep=''), color="red")+    
    annotate(geom="text", x=lastCall-4, y=.08, label=paste("Last 2% increase (",lastCall,')',sep=''), color="blue")
  
  otu_ranked<-otu_ranked
  last_call<-lastCall
  output<-list(graph,otu_ranked,lastCall,BC_ranked)
  output
}



# dia_nobac

Sven_phyto_rare_dia_nobac@sam_data$Sample<-sample_names(Sven_phyto_rare_dia_nobac)
Sven_phyto_rare_dia_nobac@sam_data$Location<-"Core"
Sven_phyto_rare_dia_nobac_core<-create_core_comm_sven(Sven_phyto_rare_dia_nobac,30000)
Sven_phyto_rare_dia_nobac_core[[1]]
if (coremethod == "elbow") {
  core_num <- Sven_phyto_rare_dia_nobac_core[[1]]$plot_env$elbow
} else {
  core_num <- Sven_phyto_rare_dia_nobac_core[[1]]$plot_env$last_call
}


dia_nobac_core<-Sven_phyto_rare_dia_nobac_core[[2]]$otu[1:core_num]
dia_nobac_comp <- microbiome::transform(Sven_phyto_rare_dia_nobac, "compositional")

dia_nobac_comp<-prune_taxa(colnames(dia_nobac_comp@otu_table@.Data)[!taxa_sums(dia_nobac_comp)==0],dia_nobac_comp)
rmeans_dia_nobac<-data.frame(RMean=colMeans(dia_nobac_comp@otu_table@.Data))
rmeans_dia_nobac$Abundance_Class<-ifelse(rownames(rmeans_dia_nobac) %in% dia_nobac_core,"Core",
                                         ifelse(rmeans_dia_nobac$RMean > RMean,"Abundant","Rare"))
rmeans_dia_nobac

rmeans_dia_nobac_DF<-as.data.frame(t(dia_nobac_comp@otu_table@.Data))
rmeans_dia_nobac_DF$Group<-rmeans_dia_nobac$Abundance_Class
dia_nobacTernarySums<-as.data.frame(t(rowsum(t(dia_nobac_comp@otu_table@.Data), rmeans_dia_nobac$Abundance_Class)))
dia_nobacTernarySums$Sample<-rownames(dia_nobacTernarySums)
dia_nobacTernarySums$CultivationStage<-dia_nobac_comp@sam_data$Cultivation_Stage
dia_nobacTernarySums$Community<-stringr::word(dia_nobacTernarySums$Sample,2,sep="_")
dia_nobacTernarySums$Community<-paste("Community",dia_nobacTernarySums$Community)
library(ggtern)
library(ggalt)
library(viridis)





dia_nobacTernarySums$CultivationStage = factor(dia_nobacTernarySums$CultivationStage,
                                           levels = c("Lag",
                                                      "Exponential",
                                                      "Stationary",
                                                      "Death"))

legend_plot<-ggplot(dia_nobacTernarySums,aes(x=CultivationStage, y=Core, color=CultivationStage)) + 
  labs(title = "",
       color ="Growth Phase") +
  #atomic_percent()   + #make ternary scales on atomic %
  geom_point(size=12,shape=16)  + 
  scale_color_viridis(discrete=TRUE)+
  theme_rgbw() + theme(
    legend.key.size = unit(3.2, 'cm'),
    legend.key.height = unit(3.2, 'cm'),
    legend.key.width = unit(3.2, 'cm'), 
    legend.title = element_text(size=40, face = "bold"), 
    legend.text = element_text(size=30))

#  geom_path(aes(group = Community), size = 1, color = "grey")

library(ggpubr)
leg <- get_legend(legend_plot)

dia_nobac_tern<-ggtern(data=dia_nobacTernarySums,mapping=aes(Abundant, Core, Rare,col=CultivationStage)) + 
  labs(title = "",
       color ="Growth Phase") +
  #atomic_percent()   + #make ternary scales on atomic %
   geom_point(size=12,shape=16)  + scale_color_viridis(discrete=TRUE)+
  theme_rgbw() +
  theme(legend.position="none",
        axis.title = element_text(size=20),
        axis.text = element_text(size = 35),
        plot.title = element_text(size = 40, face = "bold",hjust = 0.5)) + ggtitle("Diatom Only")

ggsave("", width = 12, height = 12, units = "in", device = "pdf")


Sven_phyto_rare_dia_bac@sam_data$Sample<-sample_names(Sven_phyto_rare_dia_bac)
Sven_phyto_rare_dia_bac@sam_data$Location<-"Core"
Sven_phyto_rare_dia_bac_core<-create_core_comm_sven(Sven_phyto_rare_dia_bac,30000)
dev.off()
Sven_phyto_rare_dia_bac_core[[1]] #ignore Error: object 'BC_ranked' not found shown 

if (coremethod == "elbow") {
  core_num <- Sven_phyto_rare_dia_bac_core[[1]]$plot_env$elbow
} else {
  core_num <- Sven_phyto_rare_dia_bac_core[[1]]$plot_env$last_call
}

dia_bac_core<-Sven_phyto_rare_dia_bac_core[[2]]$otu[1:core_num]
dia_bac_comp <- microbiome::transform(Sven_phyto_rare_dia_bac, "compositional")

dia_bac_comp<-prune_taxa(colnames(dia_bac_comp@otu_table@.Data)[!taxa_sums(dia_bac_comp)==0],dia_bac_comp)
rmeans_dia_bac<-data.frame(RMean=colMeans(dia_bac_comp@otu_table@.Data))
rmeans_dia_bac$Abundance_Class<-ifelse(rownames(rmeans_dia_bac) %in% dia_bac_core,"Core",
                                       ifelse(rmeans_dia_bac$RMean > RMean,"Abundant","Rare"))
rmeans_dia_bac

rmeans_dia_bac_DF<-as.data.frame(t(dia_bac_comp@otu_table@.Data))
rmeans_dia_bac_DF$Group<-rmeans_dia_bac$Abundance_Class
dia_bacTernarySums<-as.data.frame(t(rowsum(t(dia_bac_comp@otu_table@.Data), rmeans_dia_bac$Abundance_Class)))
dia_bacTernarySums$Sample<-rownames(dia_bacTernarySums)
dia_bacTernarySums$CultivationStage<-dia_bac_comp@sam_data$Cultivation_Stage
dia_bacTernarySums$Community<-stringr::word(dia_bacTernarySums$Sample,2,sep="_")
dia_bacTernarySums$Community<-paste("Community",dia_bacTernarySums$Community)

dia_bacTernarySums$CultivationStage = factor(dia_bacTernarySums$CultivationStage,
                                             levels = c("Lag",
                                                        "Exponential",
                                                        "Stationary",
                                                        "Death"))
dia_bac_tern<-ggtern(data=dia_bacTernarySums,mapping=aes(Abundant, Core, Rare,col=CultivationStage)) + 
  labs(title = "",
       color ="Growth Phase") +
  # atomic_percent()   + #make ternary scales on atomic %
   geom_point(size=12,shape=16)  + scale_color_viridis(discrete=TRUE)+
  theme_rgbw() +
  theme(legend.position="none",
        axis.title = element_text(size=20),
        axis.text = element_text(size = 35),
        plot.title = element_text(size = 40, face = "bold",hjust = 0.5)) + ggtitle("Diatom + Bacteria")

ggsave("", width = 12, height = 12, units = "in", device = "pdf")





# coc_nobac

Sven_phyto_rare_coc_nobac@sam_data$Sample<-sample_names(Sven_phyto_rare_coc_nobac)
Sven_phyto_rare_coc_nobac@sam_data$Location<-"Core"
Sven_phyto_rare_coc_nobac_core<-create_core_comm_sven(Sven_phyto_rare_coc_nobac,30000)
Sven_phyto_rare_coc_nobac_core[[1]]

if (coremethod == "elbow") {
  core_num <- Sven_phyto_rare_coc_nobac_core[[1]]$plot_env$elbow
} else {
  core_num <- Sven_phyto_rare_coc_nobac_core[[1]]$plot_env$last_call
}


coc_nobac_core<-Sven_phyto_rare_coc_nobac_core[[2]]$otu[1:core_num]
coc_nobac_comp <- microbiome::transform(Sven_phyto_rare_coc_nobac, "compositional")

coc_nobac_comp<-prune_taxa(colnames(coc_nobac_comp@otu_table@.Data)[!taxa_sums(coc_nobac_comp)==0],coc_nobac_comp)
rmeans_coc_nobac<-data.frame(RMean=colMeans(coc_nobac_comp@otu_table@.Data))
rmeans_coc_nobac$Abundance_Class<-ifelse(rownames(rmeans_coc_nobac) %in% coc_nobac_core,"Core",
                                         ifelse(rmeans_coc_nobac$RMean > RMean,"Abundant","Rare"))
rmeans_coc_nobac

rmeans_coc_nobac_DF<-as.data.frame(t(coc_nobac_comp@otu_table@.Data))
rmeans_coc_nobac_DF$Group<-rmeans_coc_nobac$Abundance_Class
coc_nobacTernarySums<-as.data.frame(t(rowsum(t(coc_nobac_comp@otu_table@.Data), rmeans_coc_nobac$Abundance_Class)))
coc_nobacTernarySums$Sample<-rownames(coc_nobacTernarySums)
coc_nobacTernarySums$CultivationStage<-coc_nobac_comp@sam_data$Cultivation_Stage
coc_nobacTernarySums$Community<-stringr::word(coc_nobacTernarySums$Sample,2,sep="_")
coc_nobacTernarySums$Community<-paste("Community",coc_nobacTernarySums$Community)

coc_nobacTernarySums$CultivationStage = factor(coc_nobacTernarySums$CultivationStage,
                                               levels = c("Lag",
                                                          "Exponential",
                                                          "Stationary",
                                                          "Death"))

coc_nobac_tern<-ggtern(data=coc_nobacTernarySums,mapping=aes(Abundant, Core, Rare,col=CultivationStage)) + 
  labs(title = "",
       color ="Growth Phase") +
  #atomic_percent()   + #make ternary scales on atomic %
   geom_point(size=12,shape=16)  + scale_color_viridis(discrete=TRUE)+
  theme_rgbw() +
  theme(legend.position="none",
        axis.title = element_text(size=20),
        axis.text = element_text(size = 35),
        plot.title = element_text(size = 40, face = "bold",hjust = 0.5)) + ggtitle("Coccolithophore Only")

ggsave("", width = 12, height = 12, units = "in", device = "pdf")


# coc_bac


Sven_phyto_rare_coc_bac@sam_data$Sample<-sample_names(Sven_phyto_rare_coc_bac)
Sven_phyto_rare_coc_bac@sam_data$Location<-"Core"
Sven_phyto_rare_coc_bac_core<-create_core_comm_sven(Sven_phyto_rare_coc_bac,30000)
Sven_phyto_rare_coc_bac_core[[1]]
if (coremethod == "elbow") {
  core_num <- Sven_phyto_rare_coc_bac_core[[1]]$plot_env$elbow
} else {
  core_num <- Sven_phyto_rare_coc_bac_core[[1]]$plot_env$last_call
}

coc_bac_core<-Sven_phyto_rare_coc_bac_core[[2]]$otu[1:core_num]
coc_bac_comp <- microbiome::transform(Sven_phyto_rare_coc_bac, "compositional")

coc_bac_comp<-prune_taxa(colnames(coc_bac_comp@otu_table@.Data)[!taxa_sums(coc_bac_comp)==0],coc_bac_comp)
rmeans_coc_bac<-data.frame(RMean=colMeans(coc_bac_comp@otu_table@.Data))
rmeans_coc_bac$Abundance_Class<-ifelse(rownames(rmeans_coc_bac) %in% coc_bac_core,"Core",
                                       ifelse(rmeans_coc_bac$RMean > RMean,"Abundant","Rare"))
rmeans_coc_bac

rmeans_coc_bac_DF<-as.data.frame(t(coc_bac_comp@otu_table@.Data))
rmeans_coc_bac_DF$Group<-rmeans_coc_bac$Abundance_Class
coc_bacTernarySums<-as.data.frame(t(rowsum(t(coc_bac_comp@otu_table@.Data), rmeans_coc_bac$Abundance_Class)))
coc_bacTernarySums$Sample<-rownames(coc_bacTernarySums)
coc_bacTernarySums$CultivationStage<-coc_bac_comp@sam_data$Cultivation_Stage
coc_bacTernarySums$Community<-stringr::word(coc_bacTernarySums$Sample,2,sep="_")
coc_bacTernarySums$Community<-paste("Community",coc_bacTernarySums$Community)

coc_bacTernarySums$CultivationStage = factor(coc_bacTernarySums$CultivationStage,
                                               levels = c("Lag",
                                                          "Exponential",
                                                          "Stationary",
                                                          "Death"))
coc_bac_tern<-ggtern(data=coc_bacTernarySums,mapping=aes(Abundant, Core, Rare,col=CultivationStage)) + 
  labs(title = "",
       color ="Growth Phase") +
  #atomic_percent()   + #make ternary scales on atomic %
   geom_point(size=12,shape=16)  + scale_color_viridis(discrete=TRUE)+
  theme_rgbw() +
  theme(legend.position="none",
        axis.title = element_text(size=20),
        axis.text = element_text(size = 35),
        plot.title = element_text(size = 40, face = "bold",hjust = 0.5))+ ggtitle("Coccolithophore + Bacteria")
ggsave("", width = 12, height =12, units = "in", device = "pdf")

# bac

Sven_phyto_rare_bac@sam_data$Sample<-sample_names(Sven_phyto_rare_bac)
Sven_phyto_rare_bac@sam_data$Location<-"Core"
Sven_phyto_rare_bac_core<-create_core_comm_sven(Sven_phyto_rare_bac,30000)
Sven_phyto_rare_bac_core[[1]]
if (coremethod == "elbow") {
  core_num <- Sven_phyto_rare_bac_core[[1]]$plot_env$elbow
} else {
  core_num <- Sven_phyto_rare_bac_core[[1]]$plot_env$last_call
}

bac_core<-Sven_phyto_rare_bac_core[[2]]$otu[1:core_num]
bac_comp <- microbiome::transform(Sven_phyto_rare_bac, "compositional")

bac_comp<-prune_taxa(colnames(bac_comp@otu_table@.Data)[!taxa_sums(bac_comp)==0],bac_comp)
rmeans_bac<-data.frame(RMean=colMeans(bac_comp@otu_table@.Data))
rmeans_bac$Abundance_Class<-ifelse(rownames(rmeans_bac) %in% bac_core,"Core",
                                   ifelse(rmeans_bac$RMean > RMean,"Abundant","Rare"))
rmeans_bac

rmeans_bac_DF<-as.data.frame(t(bac_comp@otu_table@.Data))
rmeans_bac_DF$Group<-rmeans_bac$Abundance_Class
bacTernarySums<-as.data.frame(t(rowsum(t(bac_comp@otu_table@.Data), rmeans_bac$Abundance_Class)))
bacTernarySums$Sample<-rownames(bacTernarySums)
bacTernarySums$CultivationStage<-bac_comp@sam_data$Cultivation_Stage
bacTernarySums$Community<-stringr::word(bacTernarySums$Sample,2,sep="_")
bacTernarySums$Community<-paste("Community",bacTernarySums$Community)

bacTernarySums$CultivationStage = factor(bacTernarySums$CultivationStage,
                                             levels = c("Lag",
                                                        "Exponential",
                                                        "Stationary",
                                                        "Death"))
library(ggh4x)

bac_tern<-ggtern(data=bacTernarySums,mapping=aes(Abundant, Core, Rare,col=CultivationStage)) + 
  labs(title = "",
       color ="Growth Phase") +
  #atomic_percent()   + #make ternary scales on atomic %
  geom_point(size=12,shape=16)  + scale_color_viridis(discrete=TRUE)+
  theme_rgbw() +
  theme(legend.position="none",
        axis.title = element_text(size=20),
        axis.text = element_text(size = 35),
        plot.title = element_text(size = 40, face = "bold",hjust = 0.5))+ ggtitle("Bacteria Only")
ggsave("", width = 12, height = 12, units = "in", device = "pdf")

# Saving the plots
pdf("", width = 26, height = 36)
combined_terns <- ggtern::grid.arrange(coc_bac_tern,
             coc_nobac_tern,
             dia_bac_tern,
             dia_nobac_tern,
             bac_tern, leg,  nrow = 3, ncol = 2)
dev.off(2)  

# Saving the legend separately to combine it later
# pdf("")
# legend <- ggtern::grid.arrange(leg)
# dev.off(3) 



```

#check open packages
(.packages())
##Close all but phyloseq
detachAllPackages <- function() {
  
  basic.packages <- c("package:stats","package:graphics","package:grDevices","package:utils","package:datasets","package:methods","package:base")
  
  package.list <- search()[ifelse(unlist(gregexpr("package:",search()))==1,TRUE,FALSE)]
  
  package.list <- setdiff(package.list,basic.packages)
  
  if (length(package.list)>0)  for (package in package.list) detach(package, character.only=TRUE)
  
}

detachAllPackages()

